{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "En esta sección se transferirán a tablas los transcriptomas de las especies seleccionadas.<br>\n",
    "Esto requerirá seleccionar aleatoriamente los mismos a partir de los archivos fasta.<br>\n",
    "Como primer paso, pasaremos todas las secuencias a la base de datos.<br>\n",
    "Iniciamos con los PTCs (casos negativos) en Ensembl. Nos interesan solamente las especies que también existen en CantataDB o GreeNC."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#SCRIPT PARA SECUENCIAS MUY LARGAS, QUE NO SE PODIAN CONSUMIR POR BIOMART DEBIDO A TIMEOUTS\n",
    "\n",
    "import mysql.connector\n",
    "import os\n",
    "\n",
    "#método para leer los transcriptomas de un archivo fasta en cantatadb\n",
    "def leer_transcriptomas_cantatadb(archivo, cur, id_especie):\n",
    "    cabecera = \"\"\n",
    "    transcriptoma = \"\"\n",
    "    f = open(archivo, \"r\")\n",
    "    for linea in f:\n",
    "        if linea.startswith(\">\"):\n",
    "            if transcriptoma != \"\":\n",
    "                query = \"INSERT INTO secuencias (id_especie, cod_secuencia, secuencia, flg_pct, flg_seleccionado) VALUES (%s, %s, %s, 1, NULL)\"\n",
    "                cur.execute(query, (id_especie, cabecera, transcriptoma))\n",
    "                transcriptoma = \"\"\n",
    "            cabecera = linea.rstrip(\"\\n\").lstrip(\">\")\n",
    "        else:\n",
    "            transcriptoma += linea.rstrip(\"\\n\")\n",
    "    if transcriptoma != \"\":\n",
    "        query = \"INSERT INTO secuencias (id_especie, cod_secuencia, secuencia, flg_pct, flg_seleccionado) VALUES (%s, %s, %s, 1, NULL)\"\n",
    "        cur.execute(query, (id_especie, cabecera, transcriptoma))\n",
    "        transcriptoma = \"\"\n",
    "\n",
    "conn = mysql.connector.connect(\n",
    "   host=\"localhost\",\n",
    "   database=\"tesis2\",\n",
    "   user=\"root\",\n",
    "   passwd=\"\"\n",
    ")\n",
    "\n",
    "cur = conn.cursor()\n",
    "\n",
    "directory = '.\\\\httpplants.ensembl.orgindex.html'\n",
    "\n",
    "for filename in os.listdir(directory):\n",
    "    if filename.endswith(\".fasta\"):\n",
    "        especie = os.path.splitext(filename)[0]\n",
    "        query = \"SELECT COUNT(*) FROM maestra_especies m WHERE especie IN (SELECT c1.especie FROM conteo_lncrna_especie c1 LEFT JOIN conteo_lncrna_especie c2 ON c1.especie = REPLACE(c2.especie, '_', ' ') AND c1.fuente = 'Ensembl' AND c2.fuente = 'CantataDB' LEFT JOIN conteo_lncrna_especie c3 ON c1.especie = REPLACE(c3.especie, '_', ' ') AND c1.fuente = 'Ensembl' AND c3.fuente = 'GreeNC' WHERE c1.fuente = 'Ensembl' AND (c2.fuente IS NOT NULL OR c3.fuente IS NOT NULL)) AND NOT EXISTS (SELECT 1 FROM secuencias s WHERE s.id_especie = m.id_especie AND s.flg_pct = 1) AND m.especie = REPLACE(%s, '_', ' ')\"\n",
    "        cur.execute(query, [especie])\n",
    "        result=cur.fetchone()\n",
    "        if result[0] > 0:\n",
    "            query = \"SELECT e.id_especie FROM maestra_especies e WHERE e.especie = REPLACE(%s, '_', ' ')\"\n",
    "            cur.execute(query, [especie])\n",
    "            id_especie=cur.fetchone()\n",
    "            leer_transcriptomas_cantatadb(directory + \"\\\\\" + filename, cur, id_especie[0])\n",
    "            conn.commit()\n",
    "\n",
    "conn.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "library(RMySQL)\n",
    "library(curl)\n",
    "\n",
    "dsn_database = \"tesis2\"\n",
    "dsn_hostname = \"localhost\"\n",
    "dsn_port = 3306\n",
    "dsn_uid = \"root\"\n",
    "dsn_pwd = \"\"\n",
    "\n",
    "conn = dbConnect(MySQL(), user=dsn_uid, password=dsn_pwd, dbname=dsn_database, host=dsn_hostname)\n",
    "rs = dbSendQuery(conn, \"SELECT id_especie, especie FROM maestra_especies m WHERE especie IN (SELECT c1.especie FROM conteo_lncrna_especie c1 LEFT JOIN conteo_lncrna_especie c2 ON c1.especie = REPLACE(c2.especie, '_', ' ') AND c1.fuente = 'Ensembl' AND c2.fuente = 'CantataDB' LEFT JOIN conteo_lncrna_especie c3 ON c1.especie = REPLACE(c3.especie, '_', ' ') AND c1.fuente = 'Ensembl' AND c3.fuente = 'GreeNC' WHERE c1.fuente = 'Ensembl' AND (c2.fuente IS NOT NULL OR c3.fuente IS NOT NULL)) AND NOT EXISTS (SELECT 1 FROM secuencias s WHERE s.id_especie = m.id_especie AND s.flg_pct = 1)\")\n",
    "especies = dbFetch(rs)\n",
    "dbClearResult(rs)\n",
    "\n",
    "library(biomaRt)\n",
    "library(stringr)\n",
    "ensembl_mart=useMart(\"plants_mart\", host=\"plants.ensembl.org\")\n",
    "lista_datasets = listDatasets(ensembl_mart)\n",
    "\n",
    "if (nrow(especies) > 0) {\n",
    "    for(i in 1:nrow(especies)) {\n",
    "        la_especie = especies[i,2]\n",
    "        especie = paste(\"^\", la_especie, sep=\"\")\n",
    "        dataset = lista_datasets[which(str_detect(lista_datasets$description, especie)), 1]\n",
    "        ensembl_mart = useDataset(dataset,mart=ensembl_mart)\n",
    "        genes = getBM(attributes=c('ensembl_transcript_id', 'coding'), filters = 'biotype', values = 'protein_coding', mart = ensembl_mart)\n",
    "\n",
    "        for(j in 1:nrow(genes)) {\n",
    "            query = sprintf(\"INSERT INTO secuencias (id_especie, cod_secuencia, secuencia, flg_pct, flg_seleccionado) VALUES (%d, '%s', '%s', 1, NULL)\", especies[i,1], genes[j,1], genes[j,2])\n",
    "            dbSendQuery(conn, query)\n",
    "        }\n",
    "    }\n",
    "}\n",
    "\n",
    "dbDisconnect(conn)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Ahora insertaremos las secuencias de CantataDB."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "import mysql.connector\n",
    "import os\n",
    "\n",
    "#método para leer los transcriptomas de un archivo fasta en cantatadb\n",
    "def leer_transcriptomas_cantatadb(archivo, cur, id_especie):\n",
    "    cabecera = \"\"\n",
    "    transcriptoma = \"\"\n",
    "    f = open(archivo, \"r\")\n",
    "    for linea in f:\n",
    "        if linea.startswith(\">\"):\n",
    "            if transcriptoma != \"\":\n",
    "                query = \"INSERT INTO secuencias (id_especie, cod_secuencia, secuencia, flg_pct, flg_seleccionado) VALUES (%s, %s, %s, 0, NULL)\"\n",
    "                cur.execute(query, (id_especie, cabecera, transcriptoma))\n",
    "                transcriptoma = \"\"\n",
    "            cabecera = linea.rstrip(\"\\n\").lstrip(\">\")\n",
    "        else:\n",
    "            transcriptoma += linea.rstrip(\"\\n\")\n",
    "    if transcriptoma != \"\":\n",
    "        query = \"INSERT INTO secuencias (id_especie, cod_secuencia, secuencia, flg_pct, flg_seleccionado) VALUES (%s, %s, %s, 0, NULL)\"\n",
    "        cur.execute(query, (id_especie, cabecera, transcriptoma))\n",
    "        transcriptoma = \"\"\n",
    "\n",
    "conn = mysql.connector.connect(\n",
    "   host=\"localhost\",\n",
    "   database=\"tesis2\",\n",
    "   user=\"root\",\n",
    "   passwd=\"\"\n",
    ")\n",
    "\n",
    "cur = conn.cursor()\n",
    "\n",
    "directory = '..\\\\Semana 02\\\\httpcantata.amu.edu.pldownload.php\\\\Otros'\n",
    "\n",
    "for filename in os.listdir(directory):\n",
    "    if filename.endswith(\".fasta\"):\n",
    "        especie = os.path.splitext(filename)[0].split(\"_lncrnas\")[:-1][0]\n",
    "        query = \"SELECT COUNT(*) FROM maestra_especies m WHERE especie IN (SELECT c1.especie FROM conteo_lncrna_especie c1 LEFT JOIN conteo_lncrna_especie c2 ON c1.especie = REPLACE(c2.especie, '_', ' ') AND c1.fuente = 'Ensembl' AND c2.fuente = 'CantataDB' LEFT JOIN conteo_lncrna_especie c3 ON c1.especie = REPLACE(c3.especie, '_', ' ') AND c1.fuente = 'Ensembl' AND c3.fuente = 'GreeNC' WHERE c1.fuente = 'Ensembl' AND (c2.fuente IS NOT NULL OR c3.fuente IS NOT NULL)) AND NOT EXISTS (SELECT 1 FROM secuencias s WHERE s.id_especie = m.id_especie AND s.flg_pct = 0) AND m.especie = REPLACE(%s, '_', ' ') AND EXISTS (SELECT 1 FROM especies_seleccionadas es WHERE es.especie = m.especie AND fuente = %s)\"\n",
    "        cur.execute(query, (especie, \"CantataDB\"))\n",
    "        result=cur.fetchone()\n",
    "        if result[0] > 0:\n",
    "            query = \"SELECT e.id_especie FROM maestra_especies e WHERE e.especie = REPLACE(%s, '_', ' ')\"\n",
    "            cur.execute(query, [especie])\n",
    "            id_especie=cur.fetchone()\n",
    "            leer_transcriptomas_cantatadb(directory + \"\\\\\" + filename, cur, id_especie[0])\n",
    "            conn.commit()\n",
    "\n",
    "conn.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "De forma similar, procedemos para GreeNC. Se puede reutilizar el método para leer transcriptomas de CantataDB."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "import mysql.connector\n",
    "import os\n",
    "\n",
    "#método para leer los transcriptomas de un archivo fasta en cantatadb\n",
    "def leer_transcriptomas_cantatadb(archivo, cur, id_especie):\n",
    "    cabecera = \"\"\n",
    "    transcriptoma = \"\"\n",
    "    f = open(archivo, \"r\")\n",
    "    for linea in f:\n",
    "        if linea.startswith(\">\"):\n",
    "            if transcriptoma != \"\":\n",
    "                query = \"INSERT INTO secuencias (id_especie, cod_secuencia, secuencia, flg_pct, flg_seleccionado) VALUES (%s, %s, %s, 0, NULL)\"\n",
    "                cur.execute(query, (id_especie, cabecera, transcriptoma))\n",
    "                transcriptoma = \"\"\n",
    "            cabecera = linea.rstrip(\"\\n\").lstrip(\">\")\n",
    "        else:\n",
    "            transcriptoma += linea.rstrip(\"\\n\")\n",
    "    if transcriptoma != \"\":\n",
    "        query = \"INSERT INTO secuencias (id_especie, cod_secuencia, secuencia, flg_pct, flg_seleccionado) VALUES (%s, %s, %s, 0, NULL)\"\n",
    "        cur.execute(query, (id_especie, cabecera, transcriptoma))\n",
    "        transcriptoma = \"\"\n",
    "\n",
    "conn = mysql.connector.connect(\n",
    "   host=\"localhost\",\n",
    "   database=\"tesis2\",\n",
    "   user=\"root\",\n",
    "   passwd=\"\"\n",
    ")\n",
    "\n",
    "cur = conn.cursor()\n",
    "\n",
    "directory = '.\\\\httpgreenc.sciencedesigners.com'\n",
    "\n",
    "for filename in os.listdir(directory):\n",
    "    if filename.endswith(\".fasta\"):\n",
    "        especie = os.path.splitext(filename)[0]\n",
    "        query = \"SELECT COUNT(*) FROM maestra_especies m WHERE especie IN (SELECT c1.especie FROM conteo_lncrna_especie c1 LEFT JOIN conteo_lncrna_especie c2 ON c1.especie = REPLACE(c2.especie, '_', ' ') AND c1.fuente = 'Ensembl' AND c2.fuente = 'CantataDB' LEFT JOIN conteo_lncrna_especie c3 ON c1.especie = REPLACE(c3.especie, '_', ' ') AND c1.fuente = 'Ensembl' AND c3.fuente = 'GreeNC' WHERE c1.fuente = 'Ensembl' AND (c2.fuente IS NOT NULL OR c3.fuente IS NOT NULL)) AND NOT EXISTS (SELECT 1 FROM secuencias s WHERE s.id_especie = m.id_especie AND s.flg_pct = 0) AND m.especie = REPLACE(%s, '_', ' ') AND EXISTS (SELECT 1 FROM especies_seleccionadas es WHERE es.especie = m.especie AND fuente = %s)\"\n",
    "        cur.execute(query, (especie, \"GreeNC\"))\n",
    "        result=cur.fetchone()\n",
    "        if result[0] > 0:\n",
    "            query = \"SELECT id_especie FROM maestra_especies WHERE especie = REPLACE(%s, '_', ' ')\"\n",
    "            cur.execute(query, [especie])\n",
    "            id_especie=cur.fetchone()\n",
    "            leer_transcriptomas_cantatadb(directory + \"\\\\\" + filename, cur, id_especie[0])\n",
    "            conn.commit()\n",
    "\n",
    "conn.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Conteo final."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>especie</th>\n",
       "      <th>PCT</th>\n",
       "      <th>lncRNA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Triticum aestivum</td>\n",
       "      <td>133396.0</td>\n",
       "      <td>38820.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Brassica napus</td>\n",
       "      <td>101040.0</td>\n",
       "      <td>12010.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Oryza rufipogon</td>\n",
       "      <td>47456.0</td>\n",
       "      <td>10261.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Trifolium pratense</td>\n",
       "      <td>41270.0</td>\n",
       "      <td>10179.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Physcomitrella patens</td>\n",
       "      <td>87035.0</td>\n",
       "      <td>9690.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Medicago truncatula</td>\n",
       "      <td>57585.0</td>\n",
       "      <td>9676.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Manihot esculenta</td>\n",
       "      <td>41392.0</td>\n",
       "      <td>9504.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Oryza nivara</td>\n",
       "      <td>48360.0</td>\n",
       "      <td>8955.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Brassica rapa</td>\n",
       "      <td>41025.0</td>\n",
       "      <td>8501.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Hordeum vulgare</td>\n",
       "      <td>237090.0</td>\n",
       "      <td>7970.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Arabidopsis lyrata</td>\n",
       "      <td>32667.0</td>\n",
       "      <td>7593.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Cucumis sativus</td>\n",
       "      <td>23780.0</td>\n",
       "      <td>7348.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Brassica oleracea</td>\n",
       "      <td>59220.0</td>\n",
       "      <td>7338.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Oryza barthii</td>\n",
       "      <td>41595.0</td>\n",
       "      <td>7062.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Glycine max</td>\n",
       "      <td>88412.0</td>\n",
       "      <td>6689.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Solanum tuberosum</td>\n",
       "      <td>56210.0</td>\n",
       "      <td>6680.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Corchorus capsularis</td>\n",
       "      <td>29356.0</td>\n",
       "      <td>6459.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Leersia perrieri</td>\n",
       "      <td>38961.0</td>\n",
       "      <td>6402.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Oryza brachyantha</td>\n",
       "      <td>32037.0</td>\n",
       "      <td>6004.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Amborella trichopoda</td>\n",
       "      <td>27313.0</td>\n",
       "      <td>5698.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Brachypodium distachyon</td>\n",
       "      <td>52972.0</td>\n",
       "      <td>5584.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Populus trichocarpa</td>\n",
       "      <td>73012.0</td>\n",
       "      <td>5569.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Sorghum bicolor</td>\n",
       "      <td>47110.0</td>\n",
       "      <td>5305.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Theobroma cacao</td>\n",
       "      <td>44186.0</td>\n",
       "      <td>5256.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Oryza sativa Japonica Group</td>\n",
       "      <td>42374.0</td>\n",
       "      <td>5237.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Solanum lycopersicum</td>\n",
       "      <td>33697.0</td>\n",
       "      <td>4716.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Vitis vinifera</td>\n",
       "      <td>29927.0</td>\n",
       "      <td>4542.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Arabidopsis thaliana</td>\n",
       "      <td>48332.0</td>\n",
       "      <td>4373.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Gossypium raimondii</td>\n",
       "      <td>78371.0</td>\n",
       "      <td>4216.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Setaria italica</td>\n",
       "      <td>41023.0</td>\n",
       "      <td>4208.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        especie       PCT   lncRNA\n",
       "0             Triticum aestivum  133396.0  38820.0\n",
       "1                Brassica napus  101040.0  12010.0\n",
       "2               Oryza rufipogon   47456.0  10261.0\n",
       "3            Trifolium pratense   41270.0  10179.0\n",
       "4         Physcomitrella patens   87035.0   9690.0\n",
       "5           Medicago truncatula   57585.0   9676.0\n",
       "6             Manihot esculenta   41392.0   9504.0\n",
       "7                  Oryza nivara   48360.0   8955.0\n",
       "8                 Brassica rapa   41025.0   8501.0\n",
       "9               Hordeum vulgare  237090.0   7970.0\n",
       "10           Arabidopsis lyrata   32667.0   7593.0\n",
       "11              Cucumis sativus   23780.0   7348.0\n",
       "12            Brassica oleracea   59220.0   7338.0\n",
       "13                Oryza barthii   41595.0   7062.0\n",
       "14                  Glycine max   88412.0   6689.0\n",
       "15            Solanum tuberosum   56210.0   6680.0\n",
       "16         Corchorus capsularis   29356.0   6459.0\n",
       "17             Leersia perrieri   38961.0   6402.0\n",
       "18            Oryza brachyantha   32037.0   6004.0\n",
       "19         Amborella trichopoda   27313.0   5698.0\n",
       "20      Brachypodium distachyon   52972.0   5584.0\n",
       "21          Populus trichocarpa   73012.0   5569.0\n",
       "22              Sorghum bicolor   47110.0   5305.0\n",
       "23              Theobroma cacao   44186.0   5256.0\n",
       "24  Oryza sativa Japonica Group   42374.0   5237.0\n",
       "25         Solanum lycopersicum   33697.0   4716.0\n",
       "26               Vitis vinifera   29927.0   4542.0\n",
       "27         Arabidopsis thaliana   48332.0   4373.0\n",
       "28          Gossypium raimondii   78371.0   4216.0\n",
       "29              Setaria italica   41023.0   4208.0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import mysql.connector\n",
    "import pandas as pd\n",
    "\n",
    "conn = mysql.connector.connect(\n",
    "   host=\"localhost\",\n",
    "   database=\"tesis2\",\n",
    "   user=\"root\",\n",
    "   passwd=\"\"\n",
    ")\n",
    "df = pd.read_sql_query(\"SELECT * FROM (SELECT m.especie, SUM(CASE WHEN s.flg_pct = 1 THEN 1 ELSE 0 END) AS PCT, SUM(CASE WHEN s.flg_pct = 0 THEN 1 ELSE 0 END) AS lncRNA FROM secuencias s JOIN maestra_especies m ON s.id_especie = m.id_especie GROUP BY m.especie) a WHERE a.PCT > 0 AND a.lncRNA > 0 AND a.especie NOT IN ('Zea mays') ORDER BY 3 DESC\", conn)\n",
    "display(df)\n",
    "\n",
    "conn.close()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
