{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Desde la perspectiva de la data del modelo y su arquitectura, se debe definir si se empleará un modelo end-to-end que tome como entrada los nucleótidos o un modelo con data preprocesada, en cuyo caso se deben definir features y transformar las entradas en un vector de features.\n",
    "Esta decisión depende en parte de la cantidad de información disponible. Los modelos end-to-end se prestan más para casos en los que existe una gran cantidad de información.\n",
    "Para ello se planea acceder a ensembl programáticamente para obtener el conteo de lncRNA (siempre existen más ARN codificantes que ARN no codificantes, por lo tanto el límite en la cantidad de data a utilizar vendrá por el lado del conteo de lncRNA) para cada una de las especies candidatas (para maíz, Zea mays):\n",
    "- Sorghum bicolor (Sorghum bicolor strain BTx623)\n",
    "- Setaria italica (Setaria italica strain Yugu1)\n",
    "- Oryza glaberrima (Oryza glaberrima strain CG14)\n",
    "- Oryza barthii (Oryza barthii strain IRGC:105608)\n",
    "- Oryza sativa Indica Group (Oryza sativa Indica Group strain 93-11)\n",
    "- Oryza nivara (Oryza sativa f. spontanea strain IRGC:100897)\n",
    "- Oryza rufipogon (Oryza rufipogon strain W1943)\n",
    "- Oryza sativa Japonica Group (Oryza sativa Japonica Group strain Nipponbare)\n",
    "- Oryza longistaminata (Oryza longistaminata)\n",
    "- Oryza glumipatula (Oryza glumipatula)\n",
    "- Oryza meridionalis (Oryza meridionalis strain OR44 (W2112))\n",
    "- Oryza punctata (Oryza punctata strain IRGC:105690)\n",
    "- Oryza brachyantha (Oryza brachyantha strain IRGC:101232)\n",
    "- Leersia perrieri (Leersia perrieri strain IRGC:105164)\n",
    "- Triticum aestivum (component A) (Triticum aestivum strain Chinese spring (component A))\n",
    "- Triticum dicoccoides (component A) (Triticum dicoccoides strain Zavitan (Atlit2015) (component A))\n",
    "- Triticum urartu (Triticum urartu strain G1812 (PI428198))\n",
    "- Triticum dicoccoides (component B) (Triticum dicoccoides strain Zavitan (Atlit2015) (component B))\n",
    "- Triticum aestivum (component B) (Triticum aestivum strain Chinese spring (component B))\n",
    "- Triticum dicoccoides (Triticum dicoccoides strain Zavitan (Atlit2015))\n",
    "- Triticum aestivum (Triticum aestivum strain Chinese spring)\n",
    "- Aegilops tauschii (Aegilops tauschii subsp. strangulata strain AL8/78)\n",
    "- Triticum aestivum (component D) (Triticum aestivum strain Chinese spring (component D))\n",
    "- Triticum aestivum (component U) (Triticum aestivum strain Chinese spring (component U))\n",
    "- Hordeum vulgare (Hordeum vulgare subsp. vulgare strain Morex)\n",
    "- Brachypodium distachyon (Brachypodium distachyon strain Bd21)\n",
    "- Musa acuminata (Musa acuminata subsp. malaccensis strain DH-Pahang\n",
    "\n",
    "Para ello hay que experimentar con la mejor forma de obtener esta información.\n",
    "Se utilizará el paquete de R llamado BiomaRt.\n",
    "Se experimentará obteniendo el conteo para maíz.\n",
    "\n",
    "Se toma como guía el tutorial en https://rdrr.io/bioc/biomaRt/f/vignettes/biomaRt.Rmd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "library(biomaRt)\n",
    "salida = listMarts(host=\"plants.ensembl.org\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>biomart</th>\n",
       "      <th>version</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>plants_mart</td>\n",
       "      <td>Ensembl Plants Genes 42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>plants_variations</td>\n",
       "      <td>Ensembl Plants Variations 42</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             biomart                       version\n",
       "0        plants_mart       Ensembl Plants Genes 42\n",
       "1  plants_variations  Ensembl Plants Variations 42"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "a =%R salida\n",
    "display(a)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "library(biomaRt)\n",
    "ensembl=useMart(\"plants_mart\", host=\"plants.ensembl.org\")\n",
    "salida = listDatasets(ensembl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset</th>\n",
       "      <th>description</th>\n",
       "      <th>version</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ahalleri_eg_gene</td>\n",
       "      <td>Arabidopsis halleri genes (Ahal2.2)</td>\n",
       "      <td>Ahal2.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>alyrata_eg_gene</td>\n",
       "      <td>Arabidopsis lyrata genes (v.1.0)</td>\n",
       "      <td>v.1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>atauschii_eg_gene</td>\n",
       "      <td>Aegilops tauschii genes (Aet v4.0)</td>\n",
       "      <td>Aet v4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>athaliana_eg_gene</td>\n",
       "      <td>Arabidopsis thaliana genes (TAIR10)</td>\n",
       "      <td>TAIR10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>atrichopoda_eg_gene</td>\n",
       "      <td>Amborella trichopoda genes (AMTR1.0)</td>\n",
       "      <td>AMTR1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>bdistachyon_eg_gene</td>\n",
       "      <td>Brachypodium distachyon genes (Brachypodium_di...</td>\n",
       "      <td>Brachypodium_distachyon_v3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>bnapus_eg_gene</td>\n",
       "      <td>Brassica napus genes (AST_PRJEB5043_v1)</td>\n",
       "      <td>AST_PRJEB5043_v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>boleracea_eg_gene</td>\n",
       "      <td>Brassica oleracea genes (BOL)</td>\n",
       "      <td>BOL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>brapa_eg_gene</td>\n",
       "      <td>Brassica rapa genes (Brapa_1.0)</td>\n",
       "      <td>Brapa_1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>bvulgaris_eg_gene</td>\n",
       "      <td>Beta vulgaris genes (RefBeet-1.2.2)</td>\n",
       "      <td>RefBeet-1.2.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ccapsularis_eg_gene</td>\n",
       "      <td>Corchorus capsularis genes (CCACVL1_1.0)</td>\n",
       "      <td>CCACVL1_1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ccrispus_eg_gene</td>\n",
       "      <td>Chondrus crispus genes (ASM35022v2)</td>\n",
       "      <td>ASM35022v2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>cmerolae_eg_gene</td>\n",
       "      <td>Cyanidioschyzon merolae genes (ASM9120v1)</td>\n",
       "      <td>ASM9120v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>creinhardtii_eg_gene</td>\n",
       "      <td>Chlamydomonas reinhardtii genes (Chlamydomonas...</td>\n",
       "      <td>Chlamydomonas_reinhardtii_v5.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>csativus_eg_gene</td>\n",
       "      <td>Cucumis sativus genes (ASM407v2)</td>\n",
       "      <td>ASM407v2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>dcarota_eg_gene</td>\n",
       "      <td>Daucus carota genes (ASM162521v1)</td>\n",
       "      <td>ASM162521v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>drotundata_eg_gene</td>\n",
       "      <td>Dioscorea rotundata genes (TDr96_F1_Pseudo_Chr...</td>\n",
       "      <td>TDr96_F1_Pseudo_Chromosome_v1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>gmax_eg_gene</td>\n",
       "      <td>Glycine max genes (Glycine_max_v2.1)</td>\n",
       "      <td>Glycine_max_v2.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>graimondii_eg_gene</td>\n",
       "      <td>Gossypium raimondii genes (Graimondii2_0)</td>\n",
       "      <td>Graimondii2_0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>gsulphuraria_eg_gene</td>\n",
       "      <td>Galdieria sulphuraria genes (ASM34128v1)</td>\n",
       "      <td>ASM34128v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>hannuus_eg_gene</td>\n",
       "      <td>Helianthus annuus genes (HanXRQr1.0)</td>\n",
       "      <td>HanXRQr1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>hvulgare_eg_gene</td>\n",
       "      <td>Hordeum vulgare genes (IBSC v2)</td>\n",
       "      <td>IBSC v2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>langustifolius_eg_gene</td>\n",
       "      <td>Lupinus angustifolius genes (LupAngTanjil_v1.0)</td>\n",
       "      <td>LupAngTanjil_v1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>lperrieri_eg_gene</td>\n",
       "      <td>Leersia perrieri genes (Lperr_V1.4)</td>\n",
       "      <td>Lperr_V1.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>macuminata_eg_gene</td>\n",
       "      <td>Musa acuminata genes (ASM31385v1)</td>\n",
       "      <td>ASM31385v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>mesculenta_eg_gene</td>\n",
       "      <td>Manihot esculenta genes (Manihot esculenta v6)</td>\n",
       "      <td>Manihot esculenta v6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>mtruncatula_eg_gene</td>\n",
       "      <td>Medicago truncatula genes (MedtrA17_4.0)</td>\n",
       "      <td>MedtrA17_4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>nattenuata_eg_gene</td>\n",
       "      <td>Nicotiana attenuata genes (NIATTr2)</td>\n",
       "      <td>NIATTr2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>obarthii_eg_gene</td>\n",
       "      <td>Oryza barthii genes (O.barthii_v1)</td>\n",
       "      <td>O.barthii_v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>obrachyantha_eg_gene</td>\n",
       "      <td>Oryza brachyantha genes (Oryza_brachyantha.v1.4b)</td>\n",
       "      <td>Oryza_brachyantha.v1.4b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>oglaberrima_eg_gene</td>\n",
       "      <td>Oryza glaberrima genes (Oryza_glaberrima_V1)</td>\n",
       "      <td>Oryza_glaberrima_V1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>oglumipatula_eg_gene</td>\n",
       "      <td>Oryza glumipatula genes (Oryza_glumaepatula_v1.5)</td>\n",
       "      <td>Oryza_glumaepatula_v1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>oindica_eg_gene</td>\n",
       "      <td>Oryza sativa Indica Group genes (ASM465v1)</td>\n",
       "      <td>ASM465v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>olongistaminata_eg_gene</td>\n",
       "      <td>Oryza longistaminata genes (O_longistaminata_v...</td>\n",
       "      <td>O_longistaminata_v1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>olucimarinus_eg_gene</td>\n",
       "      <td>Ostreococcus lucimarinus genes (ASM9206v1)</td>\n",
       "      <td>ASM9206v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>omeridionalis_eg_gene</td>\n",
       "      <td>Oryza meridionalis genes (Oryza_meridionalis_v...</td>\n",
       "      <td>Oryza_meridionalis_v1.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>onivara_eg_gene</td>\n",
       "      <td>Oryza nivara genes (Oryza_nivara_v1.0)</td>\n",
       "      <td>Oryza_nivara_v1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>opunctata_eg_gene</td>\n",
       "      <td>Oryza punctata genes (Oryza_punctata_v1.2)</td>\n",
       "      <td>Oryza_punctata_v1.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>orufipogon_eg_gene</td>\n",
       "      <td>Oryza rufipogon genes (OR_W1943)</td>\n",
       "      <td>OR_W1943</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>osativa_eg_gene</td>\n",
       "      <td>Oryza sativa Japonica Group genes (IRGSP-1.0)</td>\n",
       "      <td>IRGSP-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>ppatens_eg_gene</td>\n",
       "      <td>Physcomitrella patens genes (Phypa V3)</td>\n",
       "      <td>Phypa V3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>ppersica_eg_gene</td>\n",
       "      <td>Prunus persica genes (Prunus_persica_NCBIv2)</td>\n",
       "      <td>Prunus_persica_NCBIv2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>ptrichocarpa_eg_gene</td>\n",
       "      <td>Populus trichocarpa genes (Pop_tri_v3)</td>\n",
       "      <td>Pop_tri_v3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>pvulgaris_eg_gene</td>\n",
       "      <td>Phaseolus vulgaris genes (PhaVulg1_0)</td>\n",
       "      <td>PhaVulg1_0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>sbicolor_eg_gene</td>\n",
       "      <td>Sorghum bicolor genes (Sorghum_bicolor_NCBIv3)</td>\n",
       "      <td>Sorghum_bicolor_NCBIv3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>sitalica_eg_gene</td>\n",
       "      <td>Setaria italica genes (Setaria_italica_v2.0)</td>\n",
       "      <td>Setaria_italica_v2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>slycopersicum_eg_gene</td>\n",
       "      <td>Solanum lycopersicum genes (SL3.0)</td>\n",
       "      <td>SL3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>smoellendorffii_eg_gene</td>\n",
       "      <td>Selaginella moellendorffii genes (v1.0)</td>\n",
       "      <td>v1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>stuberosum_eg_gene</td>\n",
       "      <td>Solanum tuberosum genes (SolTub_3.0)</td>\n",
       "      <td>SolTub_3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>taestivum_eg_gene</td>\n",
       "      <td>Triticum aestivum genes (IWGSC)</td>\n",
       "      <td>IWGSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>tcacao_eg_gene</td>\n",
       "      <td>Theobroma cacao genes (Theobroma_cacao_20110822)</td>\n",
       "      <td>Theobroma_cacao_20110822</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>tdicoccoides_eg_gene</td>\n",
       "      <td>Triticum dicoccoides genes (WEWSeq v.1.0)</td>\n",
       "      <td>WEWSeq v.1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>tpratense_eg_gene</td>\n",
       "      <td>Trifolium pratense genes (Trpr)</td>\n",
       "      <td>Trpr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>turartu_eg_gene</td>\n",
       "      <td>Triticum urartu genes (ASM34745v1)</td>\n",
       "      <td>ASM34745v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>vangularis_eg_gene</td>\n",
       "      <td>Vigna angularis genes (Vigan1.1)</td>\n",
       "      <td>Vigan1.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>vradiata_eg_gene</td>\n",
       "      <td>Vigna radiata genes (Vradiata_ver6)</td>\n",
       "      <td>Vradiata_ver6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>vvinifera_eg_gene</td>\n",
       "      <td>Vitis vinifera genes (12X)</td>\n",
       "      <td>12X</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>zmays_eg_gene</td>\n",
       "      <td>Zea mays genes (B73 RefGen_v4)</td>\n",
       "      <td>B73 RefGen_v4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    dataset  \\\n",
       "0          ahalleri_eg_gene   \n",
       "1           alyrata_eg_gene   \n",
       "2         atauschii_eg_gene   \n",
       "3         athaliana_eg_gene   \n",
       "4       atrichopoda_eg_gene   \n",
       "5       bdistachyon_eg_gene   \n",
       "6            bnapus_eg_gene   \n",
       "7         boleracea_eg_gene   \n",
       "8             brapa_eg_gene   \n",
       "9         bvulgaris_eg_gene   \n",
       "10      ccapsularis_eg_gene   \n",
       "11         ccrispus_eg_gene   \n",
       "12         cmerolae_eg_gene   \n",
       "13     creinhardtii_eg_gene   \n",
       "14         csativus_eg_gene   \n",
       "15          dcarota_eg_gene   \n",
       "16       drotundata_eg_gene   \n",
       "17             gmax_eg_gene   \n",
       "18       graimondii_eg_gene   \n",
       "19     gsulphuraria_eg_gene   \n",
       "20          hannuus_eg_gene   \n",
       "21         hvulgare_eg_gene   \n",
       "22   langustifolius_eg_gene   \n",
       "23        lperrieri_eg_gene   \n",
       "24       macuminata_eg_gene   \n",
       "25       mesculenta_eg_gene   \n",
       "26      mtruncatula_eg_gene   \n",
       "27       nattenuata_eg_gene   \n",
       "28         obarthii_eg_gene   \n",
       "29     obrachyantha_eg_gene   \n",
       "30      oglaberrima_eg_gene   \n",
       "31     oglumipatula_eg_gene   \n",
       "32          oindica_eg_gene   \n",
       "33  olongistaminata_eg_gene   \n",
       "34     olucimarinus_eg_gene   \n",
       "35    omeridionalis_eg_gene   \n",
       "36          onivara_eg_gene   \n",
       "37        opunctata_eg_gene   \n",
       "38       orufipogon_eg_gene   \n",
       "39          osativa_eg_gene   \n",
       "40          ppatens_eg_gene   \n",
       "41         ppersica_eg_gene   \n",
       "42     ptrichocarpa_eg_gene   \n",
       "43        pvulgaris_eg_gene   \n",
       "44         sbicolor_eg_gene   \n",
       "45         sitalica_eg_gene   \n",
       "46    slycopersicum_eg_gene   \n",
       "47  smoellendorffii_eg_gene   \n",
       "48       stuberosum_eg_gene   \n",
       "49        taestivum_eg_gene   \n",
       "50           tcacao_eg_gene   \n",
       "51     tdicoccoides_eg_gene   \n",
       "52        tpratense_eg_gene   \n",
       "53          turartu_eg_gene   \n",
       "54       vangularis_eg_gene   \n",
       "55         vradiata_eg_gene   \n",
       "56        vvinifera_eg_gene   \n",
       "57            zmays_eg_gene   \n",
       "\n",
       "                                          description  \\\n",
       "0                 Arabidopsis halleri genes (Ahal2.2)   \n",
       "1                    Arabidopsis lyrata genes (v.1.0)   \n",
       "2                  Aegilops tauschii genes (Aet v4.0)   \n",
       "3                 Arabidopsis thaliana genes (TAIR10)   \n",
       "4                Amborella trichopoda genes (AMTR1.0)   \n",
       "5   Brachypodium distachyon genes (Brachypodium_di...   \n",
       "6             Brassica napus genes (AST_PRJEB5043_v1)   \n",
       "7                       Brassica oleracea genes (BOL)   \n",
       "8                     Brassica rapa genes (Brapa_1.0)   \n",
       "9                 Beta vulgaris genes (RefBeet-1.2.2)   \n",
       "10           Corchorus capsularis genes (CCACVL1_1.0)   \n",
       "11                Chondrus crispus genes (ASM35022v2)   \n",
       "12          Cyanidioschyzon merolae genes (ASM9120v1)   \n",
       "13  Chlamydomonas reinhardtii genes (Chlamydomonas...   \n",
       "14                   Cucumis sativus genes (ASM407v2)   \n",
       "15                  Daucus carota genes (ASM162521v1)   \n",
       "16  Dioscorea rotundata genes (TDr96_F1_Pseudo_Chr...   \n",
       "17               Glycine max genes (Glycine_max_v2.1)   \n",
       "18          Gossypium raimondii genes (Graimondii2_0)   \n",
       "19           Galdieria sulphuraria genes (ASM34128v1)   \n",
       "20               Helianthus annuus genes (HanXRQr1.0)   \n",
       "21                    Hordeum vulgare genes (IBSC v2)   \n",
       "22    Lupinus angustifolius genes (LupAngTanjil_v1.0)   \n",
       "23                Leersia perrieri genes (Lperr_V1.4)   \n",
       "24                  Musa acuminata genes (ASM31385v1)   \n",
       "25     Manihot esculenta genes (Manihot esculenta v6)   \n",
       "26           Medicago truncatula genes (MedtrA17_4.0)   \n",
       "27                Nicotiana attenuata genes (NIATTr2)   \n",
       "28                 Oryza barthii genes (O.barthii_v1)   \n",
       "29  Oryza brachyantha genes (Oryza_brachyantha.v1.4b)   \n",
       "30       Oryza glaberrima genes (Oryza_glaberrima_V1)   \n",
       "31  Oryza glumipatula genes (Oryza_glumaepatula_v1.5)   \n",
       "32         Oryza sativa Indica Group genes (ASM465v1)   \n",
       "33  Oryza longistaminata genes (O_longistaminata_v...   \n",
       "34         Ostreococcus lucimarinus genes (ASM9206v1)   \n",
       "35  Oryza meridionalis genes (Oryza_meridionalis_v...   \n",
       "36             Oryza nivara genes (Oryza_nivara_v1.0)   \n",
       "37         Oryza punctata genes (Oryza_punctata_v1.2)   \n",
       "38                   Oryza rufipogon genes (OR_W1943)   \n",
       "39      Oryza sativa Japonica Group genes (IRGSP-1.0)   \n",
       "40             Physcomitrella patens genes (Phypa V3)   \n",
       "41       Prunus persica genes (Prunus_persica_NCBIv2)   \n",
       "42             Populus trichocarpa genes (Pop_tri_v3)   \n",
       "43              Phaseolus vulgaris genes (PhaVulg1_0)   \n",
       "44     Sorghum bicolor genes (Sorghum_bicolor_NCBIv3)   \n",
       "45       Setaria italica genes (Setaria_italica_v2.0)   \n",
       "46                 Solanum lycopersicum genes (SL3.0)   \n",
       "47            Selaginella moellendorffii genes (v1.0)   \n",
       "48               Solanum tuberosum genes (SolTub_3.0)   \n",
       "49                    Triticum aestivum genes (IWGSC)   \n",
       "50   Theobroma cacao genes (Theobroma_cacao_20110822)   \n",
       "51          Triticum dicoccoides genes (WEWSeq v.1.0)   \n",
       "52                    Trifolium pratense genes (Trpr)   \n",
       "53                 Triticum urartu genes (ASM34745v1)   \n",
       "54                   Vigna angularis genes (Vigan1.1)   \n",
       "55                Vigna radiata genes (Vradiata_ver6)   \n",
       "56                         Vitis vinifera genes (12X)   \n",
       "57                     Zea mays genes (B73 RefGen_v4)   \n",
       "\n",
       "                            version  \n",
       "0                           Ahal2.2  \n",
       "1                             v.1.0  \n",
       "2                          Aet v4.0  \n",
       "3                            TAIR10  \n",
       "4                           AMTR1.0  \n",
       "5      Brachypodium_distachyon_v3.0  \n",
       "6                  AST_PRJEB5043_v1  \n",
       "7                               BOL  \n",
       "8                         Brapa_1.0  \n",
       "9                     RefBeet-1.2.2  \n",
       "10                      CCACVL1_1.0  \n",
       "11                       ASM35022v2  \n",
       "12                        ASM9120v1  \n",
       "13   Chlamydomonas_reinhardtii_v5.5  \n",
       "14                         ASM407v2  \n",
       "15                      ASM162521v1  \n",
       "16  TDr96_F1_Pseudo_Chromosome_v1.0  \n",
       "17                 Glycine_max_v2.1  \n",
       "18                    Graimondii2_0  \n",
       "19                       ASM34128v1  \n",
       "20                       HanXRQr1.0  \n",
       "21                          IBSC v2  \n",
       "22                LupAngTanjil_v1.0  \n",
       "23                       Lperr_V1.4  \n",
       "24                       ASM31385v1  \n",
       "25             Manihot esculenta v6  \n",
       "26                     MedtrA17_4.0  \n",
       "27                          NIATTr2  \n",
       "28                     O.barthii_v1  \n",
       "29          Oryza_brachyantha.v1.4b  \n",
       "30              Oryza_glaberrima_V1  \n",
       "31          Oryza_glumaepatula_v1.5  \n",
       "32                         ASM465v1  \n",
       "33            O_longistaminata_v1.0  \n",
       "34                        ASM9206v1  \n",
       "35          Oryza_meridionalis_v1.3  \n",
       "36                Oryza_nivara_v1.0  \n",
       "37              Oryza_punctata_v1.2  \n",
       "38                         OR_W1943  \n",
       "39                        IRGSP-1.0  \n",
       "40                         Phypa V3  \n",
       "41            Prunus_persica_NCBIv2  \n",
       "42                       Pop_tri_v3  \n",
       "43                       PhaVulg1_0  \n",
       "44           Sorghum_bicolor_NCBIv3  \n",
       "45             Setaria_italica_v2.0  \n",
       "46                            SL3.0  \n",
       "47                             v1.0  \n",
       "48                       SolTub_3.0  \n",
       "49                            IWGSC  \n",
       "50         Theobroma_cacao_20110822  \n",
       "51                     WEWSeq v.1.0  \n",
       "52                             Trpr  \n",
       "53                       ASM34745v1  \n",
       "54                         Vigan1.1  \n",
       "55                    Vradiata_ver6  \n",
       "56                              12X  \n",
       "57                    B73 RefGen_v4  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "a =%R salida\n",
    "display(a)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "especie = \"Zea mays\"\n",
    "especie = paste(\"^\", especie, sep=\"\")\n",
    "\n",
    "library(biomaRt)\n",
    "library(stringr)\n",
    "\n",
    "ensembl=useMart(\"plants_mart\", host=\"plants.ensembl.org\")\n",
    "lista_datasets = listDatasets(ensembl)\n",
    "dataset = lista_datasets[which(str_detect(lista_datasets$description, especie)), 1]\n",
    "ensembl = useDataset(dataset,mart=ensembl)\n",
    "salida = listFilters(ensembl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chromosome_name</td>\n",
       "      <td>Chromosome/scaffold name</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>start</td>\n",
       "      <td>Start</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>end</td>\n",
       "      <td>End</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>strand</td>\n",
       "      <td>Strand</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chromosomal_region</td>\n",
       "      <td>e.g. 1:100:10000:-1, 1:100000:200000:1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>with_chembl</td>\n",
       "      <td>With ChEMBL ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>with_embl</td>\n",
       "      <td>With European Nucleotide Archive ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>with_go</td>\n",
       "      <td>With GO ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>with_goslim_goa</td>\n",
       "      <td>With GOSlim GOA ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>with_protein_id</td>\n",
       "      <td>With INSDC protein ID ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>with_kegg_enzyme</td>\n",
       "      <td>With KEGG Pathway and Enzyme ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>with_maizegdb</td>\n",
       "      <td>With MaizeGDB ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>with_merops</td>\n",
       "      <td>With MEROPS - the Peptidase Database ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>with_mirbase</td>\n",
       "      <td>With miRBase ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>with_entrezgene</td>\n",
       "      <td>With NCBI gene ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>with_pdb</td>\n",
       "      <td>With PDB ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>with_plant_reactome_pathway</td>\n",
       "      <td>With Plant Reactome Pathway ID ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>with_plant_reactome_reaction</td>\n",
       "      <td>With Plant Reactome Reaction ID ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>with_refseq_dna</td>\n",
       "      <td>With RefSeq DNA ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>with_refseq_peptide</td>\n",
       "      <td>With RefSeq peptide ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>with_rfam</td>\n",
       "      <td>With RFAM ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>with_string</td>\n",
       "      <td>With STRING ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>with_trnascan_se</td>\n",
       "      <td>With tRNAScan-SE ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>with_uniparc</td>\n",
       "      <td>With UniParc ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>with_uniprotvarsplic</td>\n",
       "      <td>With UniProtKB/SpliceVariant ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>with_uniprotswissprot</td>\n",
       "      <td>With UniProtKB/Swiss-Prot ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>with_uniprotsptrembl</td>\n",
       "      <td>With UniProtKB/TrEMBL ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>ensembl_gene_id</td>\n",
       "      <td>Gene stable ID(s) [e.g. ENSRNA049437471]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>ensembl_transcript_id</td>\n",
       "      <td>Transcript stable ID(s) [e.g. ENSRNA049437471-T1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>ensembl_peptide_id</td>\n",
       "      <td>Protein stable ID(s) [e.g. ZeamMp002]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>ensembl_exon_id</td>\n",
       "      <td>Exon ID(s) [e.g. ENSRNA049437471-E1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>external_gene_name</td>\n",
       "      <td>Gene Name(s) [e.g. dull endosperm1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chembl</td>\n",
       "      <td>ChEMBL ID(s) [e.g. CHEMBL2366411]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>embl</td>\n",
       "      <td>European Nucleotide Archive ID(s) [e.g. AB007502]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>go</td>\n",
       "      <td>GO ID(s) [e.g. GO:0000002]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>goslim_goa</td>\n",
       "      <td>GOSlim GOA ID(s) [e.g. GO:0000003]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>protein_id</td>\n",
       "      <td>INSDC protein ID(s) [e.g. AAA11464.1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>kegg_enzyme</td>\n",
       "      <td>KEGG Pathway and Enzyme ID(s) [e.g. 00010+1.1....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>maizegdb</td>\n",
       "      <td>MaizeGDB ID(s) [e.g. Zm00001d000001]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>merops</td>\n",
       "      <td>MEROPS - the Peptidase Database ID(s) [e.g. A0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>mirbase_accession</td>\n",
       "      <td>miRBase accession(s) [e.g. MI0001456]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>mirbase_id</td>\n",
       "      <td>miRBase ID(s) [e.g. zma-MIR11969]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>entrezgene</td>\n",
       "      <td>NCBI gene ID(s) [e.g. 100037738]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>pdb</td>\n",
       "      <td>PDB ID(s) [e.g. 1BEA]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>plant_reactome_pathway</td>\n",
       "      <td>Plant Reactome Pathway ID(s) [e.g. R-ZMA-1119632]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>plant_reactome_reaction</td>\n",
       "      <td>Plant Reactome Reaction ID(s) [e.g. R-ZMA-1119...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>refseq_dna</td>\n",
       "      <td>RefSeq DNA ID(s) [e.g. NC_001666.2]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>refseq_peptide</td>\n",
       "      <td>RefSeq peptide ID(s) [e.g. NP_001104839.1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>rfam</td>\n",
       "      <td>RFAM ID(s) [e.g. RF00001]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>string</td>\n",
       "      <td>STRING ID(s) [e.g. 4577.AC148152.3_FGP005]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>trnascan_se</td>\n",
       "      <td>tRNAScan-SE ID(s) [e.g. tRNA-Ala]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>uniparc</td>\n",
       "      <td>UniParc ID(s) [e.g. UPI0000000067]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>uniprotvarsplic</td>\n",
       "      <td>UniProtKB/SpliceVariant ID(s) [e.g. B6TTW1-1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>uniprotswissprot</td>\n",
       "      <td>UniProtKB/Swiss-Prot ID(s) [e.g. A0A060D764]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>uniprotsptrembl</td>\n",
       "      <td>UniProtKB/TrEMBL ID(s) [e.g. A0A060CUY1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>with_affy_gpl16720</td>\n",
       "      <td>With AFFY GPL16720 probe ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>with_affy_maize</td>\n",
       "      <td>With AFFY maize probe ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>with_agilent_g2519f_016047</td>\n",
       "      <td>With AGILENT G2519F 016047 probe ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>affy_gpl16720</td>\n",
       "      <td>AFFY GPL16720 probe ID(s) [e.g. Zm066927_s_at]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>affy_maize</td>\n",
       "      <td>AFFY maize probe ID(s) [e.g. Zm.6703.1.A1_at]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>agilent_g2519f_016047</td>\n",
       "      <td>AGILENT G2519F 016047 probe ID(s) [e.g. A_92_P...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>transcript_count_greater_than</td>\n",
       "      <td>Transcript count &gt;=</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>transcript_count_less_than</td>\n",
       "      <td>Transcript count &lt;=</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>biotype</td>\n",
       "      <td>Type</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>transcript_biotype</td>\n",
       "      <td>Transcript Type</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>source</td>\n",
       "      <td>Source (gene)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>transcript_source</td>\n",
       "      <td>Source (transcript)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>go_parent_term</td>\n",
       "      <td>Parent term accession</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>go_parent_name</td>\n",
       "      <td>Parent term name</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>go_evidence_code</td>\n",
       "      <td>GO Evidence code</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>with_zmays_eg_paralog</td>\n",
       "      <td>Paralogous Zea mays Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>with_atauschii_eg_homolog</td>\n",
       "      <td>Orthologous Aegilops tauschii Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>with_atrichopoda_eg_homolog</td>\n",
       "      <td>Orthologous Amborella trichopoda Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>with_ahalleri_eg_homolog</td>\n",
       "      <td>Orthologous Arabidopsis halleri Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>with_alyrata_eg_homolog</td>\n",
       "      <td>Orthologous Arabidopsis lyrata Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>with_athaliana_eg_homolog</td>\n",
       "      <td>Orthologous Arabidopsis thaliana Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>with_bvulgaris_eg_homolog</td>\n",
       "      <td>Orthologous Beta vulgaris Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>with_bdistachyon_eg_homolog</td>\n",
       "      <td>Orthologous Brachypodium distachyon Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>with_bnapus_eg_homolog</td>\n",
       "      <td>Orthologous Brassica napus Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>with_boleracea_eg_homolog</td>\n",
       "      <td>Orthologous Brassica oleracea Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>with_brapa_eg_homolog</td>\n",
       "      <td>Orthologous Brassica rapa Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>with_creinhardtii_eg_homolog</td>\n",
       "      <td>Orthologous Chlamydomonas reinhardtii Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>with_ccrispus_eg_homolog</td>\n",
       "      <td>Orthologous Chondrus crispus Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>with_ccapsularis_eg_homolog</td>\n",
       "      <td>Orthologous Corchorus capsularis Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>with_csativus_eg_homolog</td>\n",
       "      <td>Orthologous Cucumis sativus Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>with_cmerolae_eg_homolog</td>\n",
       "      <td>Orthologous Cyanidioschyzon merolae Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>with_dcarota_eg_homolog</td>\n",
       "      <td>Orthologous Daucus carota Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>with_drotundata_eg_homolog</td>\n",
       "      <td>Orthologous Dioscorea rotundata Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>with_gsulphuraria_eg_homolog</td>\n",
       "      <td>Orthologous Galdieria sulphuraria Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>with_gmax_eg_homolog</td>\n",
       "      <td>Orthologous Glycine max Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>with_graimondii_eg_homolog</td>\n",
       "      <td>Orthologous Gossypium raimondii Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>with_hannuus_eg_homolog</td>\n",
       "      <td>Orthologous Helianthus annuus Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>with_hvulgare_eg_homolog</td>\n",
       "      <td>Orthologous Hordeum vulgare Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>with_lperrieri_eg_homolog</td>\n",
       "      <td>Orthologous Leersia perrieri Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>with_langustifolius_eg_homolog</td>\n",
       "      <td>Orthologous Lupinus angustifolius Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>with_mesculenta_eg_homolog</td>\n",
       "      <td>Orthologous Manihot esculenta Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>with_mtruncatula_eg_homolog</td>\n",
       "      <td>Orthologous Medicago truncatula Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>with_macuminata_eg_homolog</td>\n",
       "      <td>Orthologous Musa acuminata Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>with_nattenuata_eg_homolog</td>\n",
       "      <td>Orthologous Nicotiana attenuata Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>with_obarthii_eg_homolog</td>\n",
       "      <td>Orthologous Oryza barthii Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>with_obrachyantha_eg_homolog</td>\n",
       "      <td>Orthologous Oryza brachyantha Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>with_oglaberrima_eg_homolog</td>\n",
       "      <td>Orthologous Oryza glaberrima Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>with_oglumipatula_eg_homolog</td>\n",
       "      <td>Orthologous Oryza glumipatula Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>with_olongistaminata_eg_homolog</td>\n",
       "      <td>Orthologous Oryza longistaminata Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>with_omeridionalis_eg_homolog</td>\n",
       "      <td>Orthologous Oryza meridionalis Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>with_onivara_eg_homolog</td>\n",
       "      <td>Orthologous Oryza nivara Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>with_opunctata_eg_homolog</td>\n",
       "      <td>Orthologous Oryza punctata Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>with_orufipogon_eg_homolog</td>\n",
       "      <td>Orthologous Oryza rufipogon Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>with_oindica_eg_homolog</td>\n",
       "      <td>Orthologous Oryza sativa Indica Group Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>with_osativa_eg_homolog</td>\n",
       "      <td>Orthologous Oryza sativa Japonica Group Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>with_olucimarinus_eg_homolog</td>\n",
       "      <td>Orthologous Ostreococcus lucimarinus Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>with_pvulgaris_eg_homolog</td>\n",
       "      <td>Orthologous Phaseolus vulgaris Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>with_ppatens_eg_homolog</td>\n",
       "      <td>Orthologous Physcomitrella patens Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>with_ptrichocarpa_eg_homolog</td>\n",
       "      <td>Orthologous Populus trichocarpa Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>with_ppersica_eg_homolog</td>\n",
       "      <td>Orthologous Prunus persica Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>with_smoellendorffii_eg_homolog</td>\n",
       "      <td>Orthologous Selaginella moellendorffii Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116</th>\n",
       "      <td>with_sitalica_eg_homolog</td>\n",
       "      <td>Orthologous Setaria italica Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>with_slycopersicum_eg_homolog</td>\n",
       "      <td>Orthologous Solanum lycopersicum Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>with_stuberosum_eg_homolog</td>\n",
       "      <td>Orthologous Solanum tuberosum Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>with_sbicolor_eg_homolog</td>\n",
       "      <td>Orthologous Sorghum bicolor Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>with_tcacao_eg_homolog</td>\n",
       "      <td>Orthologous Theobroma cacao Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121</th>\n",
       "      <td>with_tpratense_eg_homolog</td>\n",
       "      <td>Orthologous Trifolium pratense Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122</th>\n",
       "      <td>with_taestivum_eg_homolog</td>\n",
       "      <td>Orthologous Triticum aestivum Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>123</th>\n",
       "      <td>with_tdicoccoides_eg_homolog</td>\n",
       "      <td>Orthologous Triticum dicoccoides Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>124</th>\n",
       "      <td>with_turartu_eg_homolog</td>\n",
       "      <td>Orthologous Triticum urartu Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125</th>\n",
       "      <td>with_vangularis_eg_homolog</td>\n",
       "      <td>Orthologous Vigna angularis Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126</th>\n",
       "      <td>with_vradiata_eg_homolog</td>\n",
       "      <td>Orthologous Vigna radiata Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>with_vvinifera_eg_homolog</td>\n",
       "      <td>Orthologous Vitis vinifera Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>with_interpro</td>\n",
       "      <td>With Interpro ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>with_family</td>\n",
       "      <td>With Ensembl Protein Family ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>with_cdd</td>\n",
       "      <td>With CDD ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>with_gene3d</td>\n",
       "      <td>With Gene3D ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132</th>\n",
       "      <td>with_hamap</td>\n",
       "      <td>With HAMAP ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133</th>\n",
       "      <td>with_mobidblite</td>\n",
       "      <td>With MobiDB lite</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>134</th>\n",
       "      <td>with_ncoils</td>\n",
       "      <td>With Coiled-coils (Ncoils)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>with_hmmpanther</td>\n",
       "      <td>With PANTHER ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>with_pfam</td>\n",
       "      <td>With Pfam ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>with_pirsf</td>\n",
       "      <td>With PIRSF ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>with_prints</td>\n",
       "      <td>With Prints ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>with_blastprodom</td>\n",
       "      <td>With ProDom ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>with_scanprosite</td>\n",
       "      <td>With PROSITE patterns ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141</th>\n",
       "      <td>with_pfscan</td>\n",
       "      <td>With PROSITE profiles ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>with_seg</td>\n",
       "      <td>With Low complexity (Seg)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>with_sfld</td>\n",
       "      <td>With SFLD ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144</th>\n",
       "      <td>with_signalp</td>\n",
       "      <td>With Cleavage site (Signalp)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>with_smart</td>\n",
       "      <td>With SMART ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>with_superfamily</td>\n",
       "      <td>With Superfamily ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147</th>\n",
       "      <td>with_tigrfam</td>\n",
       "      <td>With TIGRFAM ID(s)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>with_tmhmm</td>\n",
       "      <td>With Transmembrane helices</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>interpro</td>\n",
       "      <td>Interpro ID(s) [e.g. IPR000007]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>cdd</td>\n",
       "      <td>CDD ID(s) [e.g. cd00009]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>gene3d</td>\n",
       "      <td>Gene3D ID(s) [e.g. 1.10.10.10]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>hamap</td>\n",
       "      <td>HAMAP ID(s) [e.g. MF_00001]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153</th>\n",
       "      <td>hmmpanther</td>\n",
       "      <td>PANTHER ID(s) [e.g. PTHR10000]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>154</th>\n",
       "      <td>pfam</td>\n",
       "      <td>Pfam ID(s) [e.g. PF00004]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155</th>\n",
       "      <td>pirsf</td>\n",
       "      <td>PIRSF ID(s) [e.g. PIRSF000019]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156</th>\n",
       "      <td>prints</td>\n",
       "      <td>Prints ID(s) [e.g. PR00019]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157</th>\n",
       "      <td>blastprodom</td>\n",
       "      <td>ProDom ID(s) [e.g. PD000609]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>158</th>\n",
       "      <td>scanprosite</td>\n",
       "      <td>PROSITE patterns ID(s) [e.g. PS00010]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>pfscan</td>\n",
       "      <td>PROSITE profiles ID(s) [e.g. PS01031]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160</th>\n",
       "      <td>sfld</td>\n",
       "      <td>SFLD ID(s) [e.g. SFLDF00002]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>smart</td>\n",
       "      <td>SMART ID(s) [e.g. SM00014]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162</th>\n",
       "      <td>superfamily</td>\n",
       "      <td>Superfamily ID(s) [e.g. SSF100879]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>163</th>\n",
       "      <td>tigrfam</td>\n",
       "      <td>TIGRFAM ID(s) [e.g. TIGR00001]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164</th>\n",
       "      <td>germ_line_variation_source</td>\n",
       "      <td>limit to genes with germline variant data sources</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165</th>\n",
       "      <td>with_validated_snp</td>\n",
       "      <td>Variant supporting evidence</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166</th>\n",
       "      <td>so_mini_parent_name</td>\n",
       "      <td>Parent term name</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                name  \\\n",
       "0                    chromosome_name   \n",
       "1                              start   \n",
       "2                                end   \n",
       "3                             strand   \n",
       "4                 chromosomal_region   \n",
       "5                        with_chembl   \n",
       "6                          with_embl   \n",
       "7                            with_go   \n",
       "8                    with_goslim_goa   \n",
       "9                    with_protein_id   \n",
       "10                  with_kegg_enzyme   \n",
       "11                     with_maizegdb   \n",
       "12                       with_merops   \n",
       "13                      with_mirbase   \n",
       "14                   with_entrezgene   \n",
       "15                          with_pdb   \n",
       "16       with_plant_reactome_pathway   \n",
       "17      with_plant_reactome_reaction   \n",
       "18                   with_refseq_dna   \n",
       "19               with_refseq_peptide   \n",
       "20                         with_rfam   \n",
       "21                       with_string   \n",
       "22                  with_trnascan_se   \n",
       "23                      with_uniparc   \n",
       "24              with_uniprotvarsplic   \n",
       "25             with_uniprotswissprot   \n",
       "26              with_uniprotsptrembl   \n",
       "27                   ensembl_gene_id   \n",
       "28             ensembl_transcript_id   \n",
       "29                ensembl_peptide_id   \n",
       "30                   ensembl_exon_id   \n",
       "31                external_gene_name   \n",
       "32                            chembl   \n",
       "33                              embl   \n",
       "34                                go   \n",
       "35                        goslim_goa   \n",
       "36                        protein_id   \n",
       "37                       kegg_enzyme   \n",
       "38                          maizegdb   \n",
       "39                            merops   \n",
       "40                 mirbase_accession   \n",
       "41                        mirbase_id   \n",
       "42                        entrezgene   \n",
       "43                               pdb   \n",
       "44            plant_reactome_pathway   \n",
       "45           plant_reactome_reaction   \n",
       "46                        refseq_dna   \n",
       "47                    refseq_peptide   \n",
       "48                              rfam   \n",
       "49                            string   \n",
       "50                       trnascan_se   \n",
       "51                           uniparc   \n",
       "52                   uniprotvarsplic   \n",
       "53                  uniprotswissprot   \n",
       "54                   uniprotsptrembl   \n",
       "55                with_affy_gpl16720   \n",
       "56                   with_affy_maize   \n",
       "57        with_agilent_g2519f_016047   \n",
       "58                     affy_gpl16720   \n",
       "59                        affy_maize   \n",
       "60             agilent_g2519f_016047   \n",
       "61     transcript_count_greater_than   \n",
       "62        transcript_count_less_than   \n",
       "63                           biotype   \n",
       "64                transcript_biotype   \n",
       "65                            source   \n",
       "66                 transcript_source   \n",
       "67                    go_parent_term   \n",
       "68                    go_parent_name   \n",
       "69                  go_evidence_code   \n",
       "70             with_zmays_eg_paralog   \n",
       "71         with_atauschii_eg_homolog   \n",
       "72       with_atrichopoda_eg_homolog   \n",
       "73          with_ahalleri_eg_homolog   \n",
       "74           with_alyrata_eg_homolog   \n",
       "75         with_athaliana_eg_homolog   \n",
       "76         with_bvulgaris_eg_homolog   \n",
       "77       with_bdistachyon_eg_homolog   \n",
       "78            with_bnapus_eg_homolog   \n",
       "79         with_boleracea_eg_homolog   \n",
       "80             with_brapa_eg_homolog   \n",
       "81      with_creinhardtii_eg_homolog   \n",
       "82          with_ccrispus_eg_homolog   \n",
       "83       with_ccapsularis_eg_homolog   \n",
       "84          with_csativus_eg_homolog   \n",
       "85          with_cmerolae_eg_homolog   \n",
       "86           with_dcarota_eg_homolog   \n",
       "87        with_drotundata_eg_homolog   \n",
       "88      with_gsulphuraria_eg_homolog   \n",
       "89              with_gmax_eg_homolog   \n",
       "90        with_graimondii_eg_homolog   \n",
       "91           with_hannuus_eg_homolog   \n",
       "92          with_hvulgare_eg_homolog   \n",
       "93         with_lperrieri_eg_homolog   \n",
       "94    with_langustifolius_eg_homolog   \n",
       "95        with_mesculenta_eg_homolog   \n",
       "96       with_mtruncatula_eg_homolog   \n",
       "97        with_macuminata_eg_homolog   \n",
       "98        with_nattenuata_eg_homolog   \n",
       "99          with_obarthii_eg_homolog   \n",
       "100     with_obrachyantha_eg_homolog   \n",
       "101      with_oglaberrima_eg_homolog   \n",
       "102     with_oglumipatula_eg_homolog   \n",
       "103  with_olongistaminata_eg_homolog   \n",
       "104    with_omeridionalis_eg_homolog   \n",
       "105          with_onivara_eg_homolog   \n",
       "106        with_opunctata_eg_homolog   \n",
       "107       with_orufipogon_eg_homolog   \n",
       "108          with_oindica_eg_homolog   \n",
       "109          with_osativa_eg_homolog   \n",
       "110     with_olucimarinus_eg_homolog   \n",
       "111        with_pvulgaris_eg_homolog   \n",
       "112          with_ppatens_eg_homolog   \n",
       "113     with_ptrichocarpa_eg_homolog   \n",
       "114         with_ppersica_eg_homolog   \n",
       "115  with_smoellendorffii_eg_homolog   \n",
       "116         with_sitalica_eg_homolog   \n",
       "117    with_slycopersicum_eg_homolog   \n",
       "118       with_stuberosum_eg_homolog   \n",
       "119         with_sbicolor_eg_homolog   \n",
       "120           with_tcacao_eg_homolog   \n",
       "121        with_tpratense_eg_homolog   \n",
       "122        with_taestivum_eg_homolog   \n",
       "123     with_tdicoccoides_eg_homolog   \n",
       "124          with_turartu_eg_homolog   \n",
       "125       with_vangularis_eg_homolog   \n",
       "126         with_vradiata_eg_homolog   \n",
       "127        with_vvinifera_eg_homolog   \n",
       "128                    with_interpro   \n",
       "129                      with_family   \n",
       "130                         with_cdd   \n",
       "131                      with_gene3d   \n",
       "132                       with_hamap   \n",
       "133                  with_mobidblite   \n",
       "134                      with_ncoils   \n",
       "135                  with_hmmpanther   \n",
       "136                        with_pfam   \n",
       "137                       with_pirsf   \n",
       "138                      with_prints   \n",
       "139                 with_blastprodom   \n",
       "140                 with_scanprosite   \n",
       "141                      with_pfscan   \n",
       "142                         with_seg   \n",
       "143                        with_sfld   \n",
       "144                     with_signalp   \n",
       "145                       with_smart   \n",
       "146                 with_superfamily   \n",
       "147                     with_tigrfam   \n",
       "148                       with_tmhmm   \n",
       "149                         interpro   \n",
       "150                              cdd   \n",
       "151                           gene3d   \n",
       "152                            hamap   \n",
       "153                       hmmpanther   \n",
       "154                             pfam   \n",
       "155                            pirsf   \n",
       "156                           prints   \n",
       "157                      blastprodom   \n",
       "158                      scanprosite   \n",
       "159                           pfscan   \n",
       "160                             sfld   \n",
       "161                            smart   \n",
       "162                      superfamily   \n",
       "163                          tigrfam   \n",
       "164       germ_line_variation_source   \n",
       "165               with_validated_snp   \n",
       "166              so_mini_parent_name   \n",
       "\n",
       "                                           description  \n",
       "0                             Chromosome/scaffold name  \n",
       "1                                                Start  \n",
       "2                                                  End  \n",
       "3                                               Strand  \n",
       "4               e.g. 1:100:10000:-1, 1:100000:200000:1  \n",
       "5                                    With ChEMBL ID(s)  \n",
       "6               With European Nucleotide Archive ID(s)  \n",
       "7                                        With GO ID(s)  \n",
       "8                                With GOSlim GOA ID(s)  \n",
       "9                          With INSDC protein ID ID(s)  \n",
       "10                  With KEGG Pathway and Enzyme ID(s)  \n",
       "11                                 With MaizeGDB ID(s)  \n",
       "12          With MEROPS - the Peptidase Database ID(s)  \n",
       "13                                  With miRBase ID(s)  \n",
       "14                                With NCBI gene ID(s)  \n",
       "15                                      With PDB ID(s)  \n",
       "16                With Plant Reactome Pathway ID ID(s)  \n",
       "17               With Plant Reactome Reaction ID ID(s)  \n",
       "18                               With RefSeq DNA ID(s)  \n",
       "19                           With RefSeq peptide ID(s)  \n",
       "20                                     With RFAM ID(s)  \n",
       "21                                   With STRING ID(s)  \n",
       "22                              With tRNAScan-SE ID(s)  \n",
       "23                                  With UniParc ID(s)  \n",
       "24                  With UniProtKB/SpliceVariant ID(s)  \n",
       "25                     With UniProtKB/Swiss-Prot ID(s)  \n",
       "26                         With UniProtKB/TrEMBL ID(s)  \n",
       "27            Gene stable ID(s) [e.g. ENSRNA049437471]  \n",
       "28   Transcript stable ID(s) [e.g. ENSRNA049437471-T1]  \n",
       "29               Protein stable ID(s) [e.g. ZeamMp002]  \n",
       "30                Exon ID(s) [e.g. ENSRNA049437471-E1]  \n",
       "31                 Gene Name(s) [e.g. dull endosperm1]  \n",
       "32                   ChEMBL ID(s) [e.g. CHEMBL2366411]  \n",
       "33   European Nucleotide Archive ID(s) [e.g. AB007502]  \n",
       "34                          GO ID(s) [e.g. GO:0000002]  \n",
       "35                  GOSlim GOA ID(s) [e.g. GO:0000003]  \n",
       "36               INSDC protein ID(s) [e.g. AAA11464.1]  \n",
       "37   KEGG Pathway and Enzyme ID(s) [e.g. 00010+1.1....  \n",
       "38                MaizeGDB ID(s) [e.g. Zm00001d000001]  \n",
       "39   MEROPS - the Peptidase Database ID(s) [e.g. A0...  \n",
       "40               miRBase accession(s) [e.g. MI0001456]  \n",
       "41                   miRBase ID(s) [e.g. zma-MIR11969]  \n",
       "42                    NCBI gene ID(s) [e.g. 100037738]  \n",
       "43                               PDB ID(s) [e.g. 1BEA]  \n",
       "44   Plant Reactome Pathway ID(s) [e.g. R-ZMA-1119632]  \n",
       "45   Plant Reactome Reaction ID(s) [e.g. R-ZMA-1119...  \n",
       "46                 RefSeq DNA ID(s) [e.g. NC_001666.2]  \n",
       "47          RefSeq peptide ID(s) [e.g. NP_001104839.1]  \n",
       "48                           RFAM ID(s) [e.g. RF00001]  \n",
       "49          STRING ID(s) [e.g. 4577.AC148152.3_FGP005]  \n",
       "50                   tRNAScan-SE ID(s) [e.g. tRNA-Ala]  \n",
       "51                  UniParc ID(s) [e.g. UPI0000000067]  \n",
       "52       UniProtKB/SpliceVariant ID(s) [e.g. B6TTW1-1]  \n",
       "53        UniProtKB/Swiss-Prot ID(s) [e.g. A0A060D764]  \n",
       "54            UniProtKB/TrEMBL ID(s) [e.g. A0A060CUY1]  \n",
       "55                      With AFFY GPL16720 probe ID(s)  \n",
       "56                         With AFFY maize probe ID(s)  \n",
       "57              With AGILENT G2519F 016047 probe ID(s)  \n",
       "58      AFFY GPL16720 probe ID(s) [e.g. Zm066927_s_at]  \n",
       "59       AFFY maize probe ID(s) [e.g. Zm.6703.1.A1_at]  \n",
       "60   AGILENT G2519F 016047 probe ID(s) [e.g. A_92_P...  \n",
       "61                                 Transcript count >=  \n",
       "62                                 Transcript count <=  \n",
       "63                                                Type  \n",
       "64                                     Transcript Type  \n",
       "65                                       Source (gene)  \n",
       "66                                 Source (transcript)  \n",
       "67                               Parent term accession  \n",
       "68                                    Parent term name  \n",
       "69                                    GO Evidence code  \n",
       "70                           Paralogous Zea mays Genes  \n",
       "71                 Orthologous Aegilops tauschii Genes  \n",
       "72              Orthologous Amborella trichopoda Genes  \n",
       "73               Orthologous Arabidopsis halleri Genes  \n",
       "74                Orthologous Arabidopsis lyrata Genes  \n",
       "75              Orthologous Arabidopsis thaliana Genes  \n",
       "76                     Orthologous Beta vulgaris Genes  \n",
       "77           Orthologous Brachypodium distachyon Genes  \n",
       "78                    Orthologous Brassica napus Genes  \n",
       "79                 Orthologous Brassica oleracea Genes  \n",
       "80                     Orthologous Brassica rapa Genes  \n",
       "81         Orthologous Chlamydomonas reinhardtii Genes  \n",
       "82                  Orthologous Chondrus crispus Genes  \n",
       "83              Orthologous Corchorus capsularis Genes  \n",
       "84                   Orthologous Cucumis sativus Genes  \n",
       "85           Orthologous Cyanidioschyzon merolae Genes  \n",
       "86                     Orthologous Daucus carota Genes  \n",
       "87               Orthologous Dioscorea rotundata Genes  \n",
       "88             Orthologous Galdieria sulphuraria Genes  \n",
       "89                       Orthologous Glycine max Genes  \n",
       "90               Orthologous Gossypium raimondii Genes  \n",
       "91                 Orthologous Helianthus annuus Genes  \n",
       "92                   Orthologous Hordeum vulgare Genes  \n",
       "93                  Orthologous Leersia perrieri Genes  \n",
       "94             Orthologous Lupinus angustifolius Genes  \n",
       "95                 Orthologous Manihot esculenta Genes  \n",
       "96               Orthologous Medicago truncatula Genes  \n",
       "97                    Orthologous Musa acuminata Genes  \n",
       "98               Orthologous Nicotiana attenuata Genes  \n",
       "99                     Orthologous Oryza barthii Genes  \n",
       "100                Orthologous Oryza brachyantha Genes  \n",
       "101                 Orthologous Oryza glaberrima Genes  \n",
       "102                Orthologous Oryza glumipatula Genes  \n",
       "103             Orthologous Oryza longistaminata Genes  \n",
       "104               Orthologous Oryza meridionalis Genes  \n",
       "105                     Orthologous Oryza nivara Genes  \n",
       "106                   Orthologous Oryza punctata Genes  \n",
       "107                  Orthologous Oryza rufipogon Genes  \n",
       "108        Orthologous Oryza sativa Indica Group Genes  \n",
       "109      Orthologous Oryza sativa Japonica Group Genes  \n",
       "110         Orthologous Ostreococcus lucimarinus Genes  \n",
       "111               Orthologous Phaseolus vulgaris Genes  \n",
       "112            Orthologous Physcomitrella patens Genes  \n",
       "113              Orthologous Populus trichocarpa Genes  \n",
       "114                   Orthologous Prunus persica Genes  \n",
       "115       Orthologous Selaginella moellendorffii Genes  \n",
       "116                  Orthologous Setaria italica Genes  \n",
       "117             Orthologous Solanum lycopersicum Genes  \n",
       "118                Orthologous Solanum tuberosum Genes  \n",
       "119                  Orthologous Sorghum bicolor Genes  \n",
       "120                  Orthologous Theobroma cacao Genes  \n",
       "121               Orthologous Trifolium pratense Genes  \n",
       "122                Orthologous Triticum aestivum Genes  \n",
       "123             Orthologous Triticum dicoccoides Genes  \n",
       "124                  Orthologous Triticum urartu Genes  \n",
       "125                  Orthologous Vigna angularis Genes  \n",
       "126                    Orthologous Vigna radiata Genes  \n",
       "127                   Orthologous Vitis vinifera Genes  \n",
       "128                                With Interpro ID(s)  \n",
       "129                  With Ensembl Protein Family ID(s)  \n",
       "130                                     With CDD ID(s)  \n",
       "131                                  With Gene3D ID(s)  \n",
       "132                                   With HAMAP ID(s)  \n",
       "133                                   With MobiDB lite  \n",
       "134                         With Coiled-coils (Ncoils)  \n",
       "135                                 With PANTHER ID(s)  \n",
       "136                                    With Pfam ID(s)  \n",
       "137                                   With PIRSF ID(s)  \n",
       "138                                  With Prints ID(s)  \n",
       "139                                  With ProDom ID(s)  \n",
       "140                        With PROSITE patterns ID(s)  \n",
       "141                        With PROSITE profiles ID(s)  \n",
       "142                          With Low complexity (Seg)  \n",
       "143                                    With SFLD ID(s)  \n",
       "144                       With Cleavage site (Signalp)  \n",
       "145                                   With SMART ID(s)  \n",
       "146                             With Superfamily ID(s)  \n",
       "147                                 With TIGRFAM ID(s)  \n",
       "148                         With Transmembrane helices  \n",
       "149                    Interpro ID(s) [e.g. IPR000007]  \n",
       "150                           CDD ID(s) [e.g. cd00009]  \n",
       "151                     Gene3D ID(s) [e.g. 1.10.10.10]  \n",
       "152                        HAMAP ID(s) [e.g. MF_00001]  \n",
       "153                     PANTHER ID(s) [e.g. PTHR10000]  \n",
       "154                          Pfam ID(s) [e.g. PF00004]  \n",
       "155                     PIRSF ID(s) [e.g. PIRSF000019]  \n",
       "156                        Prints ID(s) [e.g. PR00019]  \n",
       "157                       ProDom ID(s) [e.g. PD000609]  \n",
       "158              PROSITE patterns ID(s) [e.g. PS00010]  \n",
       "159              PROSITE profiles ID(s) [e.g. PS01031]  \n",
       "160                       SFLD ID(s) [e.g. SFLDF00002]  \n",
       "161                         SMART ID(s) [e.g. SM00014]  \n",
       "162                 Superfamily ID(s) [e.g. SSF100879]  \n",
       "163                     TIGRFAM ID(s) [e.g. TIGR00001]  \n",
       "164  limit to genes with germline variant data sources  \n",
       "165                        Variant supporting evidence  \n",
       "166                                   Parent term name  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "pd.set_option('display.max_rows', 200)\n",
    "a =%R salida\n",
    "display(a)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "especie = \"Zea mays\"\n",
    "especie = paste(\"^\", especie, sep=\"\")\n",
    "\n",
    "library(biomaRt)\n",
    "library(stringr)\n",
    "\n",
    "ensembl=useMart(\"plants_mart\", host=\"plants.ensembl.org\")\n",
    "lista_datasets = listDatasets(ensembl)\n",
    "dataset = lista_datasets[which(str_detect(lista_datasets$description, especie)), 1]\n",
    "ensembl = useDataset(dataset,mart=ensembl)\n",
    "salida = listAttributes(ensembl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>description</th>\n",
       "      <th>page</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ensembl_gene_id</td>\n",
       "      <td>Gene stable ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ensembl_transcript_id</td>\n",
       "      <td>Transcript stable ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ensembl_peptide_id</td>\n",
       "      <td>Protein stable ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ensembl_exon_id</td>\n",
       "      <td>Exon stable ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>description</td>\n",
       "      <td>Gene description</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chromosome_name</td>\n",
       "      <td>Chromosome/scaffold name</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>start_position</td>\n",
       "      <td>Gene start (bp)</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>end_position</td>\n",
       "      <td>Gene end (bp)</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>strand</td>\n",
       "      <td>Strand</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>band</td>\n",
       "      <td>Karyotype band</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>transcript_start</td>\n",
       "      <td>Transcript start (bp)</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>transcript_end</td>\n",
       "      <td>Transcript end (bp)</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>transcription_start_site</td>\n",
       "      <td>Transcription start site (TSS)</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>transcript_length</td>\n",
       "      <td>Transcript length (including UTRs and CDS)</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>external_gene_name</td>\n",
       "      <td>Gene name</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>external_gene_source</td>\n",
       "      <td>Source of gene name</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>transcript_count</td>\n",
       "      <td>Transcript count</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>percentage_gene_gc_content</td>\n",
       "      <td>Gene % GC content</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>gene_biotype</td>\n",
       "      <td>Gene type</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>transcript_biotype</td>\n",
       "      <td>Transcript type</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>source</td>\n",
       "      <td>Source (gene)</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>transcript_source</td>\n",
       "      <td>Source (transcript)</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>go_id</td>\n",
       "      <td>GO term accession</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>name_1006</td>\n",
       "      <td>GO term name</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>definition_1006</td>\n",
       "      <td>GO term definition</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>go_linkage_type</td>\n",
       "      <td>GO term evidence code</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>namespace_1003</td>\n",
       "      <td>GO domain</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>goslim_goa_accession</td>\n",
       "      <td>GOSlim GOA Accession(s)</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>goslim_goa_description</td>\n",
       "      <td>GOSlim GOA Description</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chembl</td>\n",
       "      <td>ChEMBL ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>embl</td>\n",
       "      <td>European Nucleotide Archive ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>protein_id</td>\n",
       "      <td>INSDC protein ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>kegg_enzyme</td>\n",
       "      <td>KEGG Pathway and Enzyme ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>maizegdb</td>\n",
       "      <td>MaizeGDB ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>merops</td>\n",
       "      <td>MEROPS - the Peptidase Database ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>mirbase_accession</td>\n",
       "      <td>miRBase accession</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>mirbase_id</td>\n",
       "      <td>miRBase ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>entrezgene</td>\n",
       "      <td>NCBI gene ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>pdb</td>\n",
       "      <td>PDB ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>plant_reactome_pathway</td>\n",
       "      <td>Plant Reactome Pathway ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>plant_reactome_reaction</td>\n",
       "      <td>Plant Reactome Reaction ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>refseq_dna</td>\n",
       "      <td>RefSeq DNA ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>refseq_peptide</td>\n",
       "      <td>RefSeq peptide ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>rfam</td>\n",
       "      <td>RFAM ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>string</td>\n",
       "      <td>STRING ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>trnascan_se</td>\n",
       "      <td>tRNAScan-SE ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>uniparc</td>\n",
       "      <td>UniParc ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>uniprotvarsplic</td>\n",
       "      <td>UniProtKB/SpliceVariant ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>uniprotswissprot</td>\n",
       "      <td>UniProtKB/Swiss-Prot ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>uniprotsptrembl</td>\n",
       "      <td>UniProtKB/TrEMBL ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>affy_gpl16720</td>\n",
       "      <td>AFFY GPL16720 probe</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>affy_maize</td>\n",
       "      <td>AFFY maize probe</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>agilent_g2519f_016047</td>\n",
       "      <td>AGILENT G2519F 016047 probe</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>cdd</td>\n",
       "      <td>CDD ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>cdd_start</td>\n",
       "      <td>CDD start</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>cdd_end</td>\n",
       "      <td>CDD end</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>gene3d</td>\n",
       "      <td>Gene3D ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>gene3d_start</td>\n",
       "      <td>Gene3D start</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>gene3d_end</td>\n",
       "      <td>Gene3D end</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>hamap</td>\n",
       "      <td>HAMAP ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>hamap_start</td>\n",
       "      <td>HAMAP start</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>hamap_end</td>\n",
       "      <td>HAMAP end</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>hmmpanther</td>\n",
       "      <td>PANTHER ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>hmmpanther_start</td>\n",
       "      <td>PANTHER start</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>hmmpanther_end</td>\n",
       "      <td>PANTHER end</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>pfam</td>\n",
       "      <td>Pfam ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>pfam_start</td>\n",
       "      <td>Pfam start</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>pfam_end</td>\n",
       "      <td>Pfam end</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>pirsf</td>\n",
       "      <td>PIRSF ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>pirsf_start</td>\n",
       "      <td>PIRSF start</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>pirsf_end</td>\n",
       "      <td>PIRSF end</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>prints</td>\n",
       "      <td>Prints ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>prints_start</td>\n",
       "      <td>Prints start</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>prints_end</td>\n",
       "      <td>Prints end</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>blastprodom</td>\n",
       "      <td>ProDom ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>blastprodom_start</td>\n",
       "      <td>ProDom start</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>blastprodom_end</td>\n",
       "      <td>ProDom end</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>scanprosite</td>\n",
       "      <td>PROSITE patterns ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>scanprosite_start</td>\n",
       "      <td>PROSITE patterns start</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>scanprosite_end</td>\n",
       "      <td>PROSITE patterns end</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>pfscan</td>\n",
       "      <td>PROSITE profiles ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>pfscan_start</td>\n",
       "      <td>PROSITE profiles start</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>pfscan_end</td>\n",
       "      <td>PROSITE profiles end</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>sfld</td>\n",
       "      <td>SFLD ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>sfld_start</td>\n",
       "      <td>SFLD start</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>sfld_end</td>\n",
       "      <td>SFLD end</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>smart</td>\n",
       "      <td>SMART ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>smart_start</td>\n",
       "      <td>SMART start</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>smart_end</td>\n",
       "      <td>SMART end</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>superfamily</td>\n",
       "      <td>Superfamily ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>superfamily_start</td>\n",
       "      <td>Superfamily start</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>superfamily_end</td>\n",
       "      <td>Superfamily end</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>tigrfam</td>\n",
       "      <td>TIGRFAM ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>tigrfam_start</td>\n",
       "      <td>TIGRFAM start</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>tigrfam_end</td>\n",
       "      <td>TIGRFAM end</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>interpro</td>\n",
       "      <td>Interpro ID</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>interpro_short_description</td>\n",
       "      <td>Interpro Short Description</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>interpro_description</td>\n",
       "      <td>Interpro Description</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>interpro_start</td>\n",
       "      <td>Interpro start</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>interpro_end</td>\n",
       "      <td>Interpro end</td>\n",
       "      <td>feature_page</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1080</th>\n",
       "      <td>zmays_eg_paralog_canonical_transcript_protein</td>\n",
       "      <td>Paralogue query protein or transcript ID</td>\n",
       "      <td>homologs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1081</th>\n",
       "      <td>zmays_eg_paralog_subtype</td>\n",
       "      <td>Paralogue last common ancestor with Zea mays</td>\n",
       "      <td>homologs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1082</th>\n",
       "      <td>zmays_eg_paralog_orthology_type</td>\n",
       "      <td>Zea mays paralogue homology type</td>\n",
       "      <td>homologs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1083</th>\n",
       "      <td>zmays_eg_paralog_perc_id</td>\n",
       "      <td>Paralogue %id. target Zea mays gene identical ...</td>\n",
       "      <td>homologs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1084</th>\n",
       "      <td>zmays_eg_paralog_perc_id_r1</td>\n",
       "      <td>Paralogue %id. query gene identical to target ...</td>\n",
       "      <td>homologs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1085</th>\n",
       "      <td>zmays_eg_paralog_dn</td>\n",
       "      <td>Paralogue dN with Zea mays</td>\n",
       "      <td>homologs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1086</th>\n",
       "      <td>zmays_eg_paralog_ds</td>\n",
       "      <td>Paralogue dS with Zea mays</td>\n",
       "      <td>homologs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1087</th>\n",
       "      <td>zmays_eg_paralog_paralogy_confidence</td>\n",
       "      <td>Zea mays paralogy confidence [0 low, 1 high]</td>\n",
       "      <td>homologs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1088</th>\n",
       "      <td>ensembl_gene_id</td>\n",
       "      <td>Gene stable ID</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1089</th>\n",
       "      <td>ensembl_transcript_id</td>\n",
       "      <td>Transcript stable ID</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1090</th>\n",
       "      <td>ensembl_peptide_id</td>\n",
       "      <td>Protein stable ID</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1091</th>\n",
       "      <td>chromosome_name</td>\n",
       "      <td>Chromosome/scaffold name</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1092</th>\n",
       "      <td>start_position</td>\n",
       "      <td>Gene start (bp)</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1093</th>\n",
       "      <td>end_position</td>\n",
       "      <td>Gene end (bp)</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1094</th>\n",
       "      <td>strand</td>\n",
       "      <td>Strand</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1095</th>\n",
       "      <td>band</td>\n",
       "      <td>Karyotype band</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>external_gene_name</td>\n",
       "      <td>Gene name</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1097</th>\n",
       "      <td>external_gene_source</td>\n",
       "      <td>Source of gene name</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1098</th>\n",
       "      <td>transcript_count</td>\n",
       "      <td>Transcript count</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1099</th>\n",
       "      <td>percentage_gene_gc_content</td>\n",
       "      <td>Gene % GC content</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1100</th>\n",
       "      <td>description</td>\n",
       "      <td>Gene description</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1101</th>\n",
       "      <td>variation_name</td>\n",
       "      <td>Variant name</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1102</th>\n",
       "      <td>germ_line_variation_source</td>\n",
       "      <td>Variant source</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1103</th>\n",
       "      <td>source_description</td>\n",
       "      <td>Variant source description</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1104</th>\n",
       "      <td>allele</td>\n",
       "      <td>Variant alleles</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1105</th>\n",
       "      <td>validated</td>\n",
       "      <td>Variant supporting evidence</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1106</th>\n",
       "      <td>mapweight</td>\n",
       "      <td>Mapweight</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1107</th>\n",
       "      <td>minor_allele</td>\n",
       "      <td>Minor allele</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1108</th>\n",
       "      <td>minor_allele_freq</td>\n",
       "      <td>Minor allele frequency</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1109</th>\n",
       "      <td>minor_allele_count</td>\n",
       "      <td>Minor allele count</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1110</th>\n",
       "      <td>clinical_significance</td>\n",
       "      <td>Clinical significance</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1111</th>\n",
       "      <td>transcript_location</td>\n",
       "      <td>Transcript location (bp)</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1112</th>\n",
       "      <td>snp_chromosome_strand</td>\n",
       "      <td>Variant chromosome Strand</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1113</th>\n",
       "      <td>peptide_location</td>\n",
       "      <td>Protein location (aa)</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1114</th>\n",
       "      <td>chromosome_start</td>\n",
       "      <td>chromosome/scaffold position start (bp)</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1115</th>\n",
       "      <td>chromosome_end</td>\n",
       "      <td>Chromosome/scaffold position end (bp)</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1116</th>\n",
       "      <td>polyphen_prediction_2076</td>\n",
       "      <td>PolyPhen prediction</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1117</th>\n",
       "      <td>polyphen_score_2076</td>\n",
       "      <td>PolyPhen score</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1118</th>\n",
       "      <td>sift_prediction_2076</td>\n",
       "      <td>SIFT prediction</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1119</th>\n",
       "      <td>sift_score_2076</td>\n",
       "      <td>SIFT score</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1120</th>\n",
       "      <td>distance_to_transcript_2076</td>\n",
       "      <td>Distance to transcript</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1121</th>\n",
       "      <td>cds_start_2076</td>\n",
       "      <td>CDS start</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1122</th>\n",
       "      <td>cds_end_2076</td>\n",
       "      <td>CDS end</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1123</th>\n",
       "      <td>peptide_shift</td>\n",
       "      <td>Protein allele</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1124</th>\n",
       "      <td>synonymous_status</td>\n",
       "      <td>Variant consequence</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1125</th>\n",
       "      <td>allele_string_2076</td>\n",
       "      <td>Consequence specific allele</td>\n",
       "      <td>snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1126</th>\n",
       "      <td>transcript_exon_intron</td>\n",
       "      <td>Unspliced (Transcript)</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1127</th>\n",
       "      <td>gene_exon_intron</td>\n",
       "      <td>Unspliced (Gene)</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128</th>\n",
       "      <td>transcript_flank</td>\n",
       "      <td>Flank (Transcript)</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1129</th>\n",
       "      <td>gene_flank</td>\n",
       "      <td>Flank (Gene)</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1130</th>\n",
       "      <td>coding_transcript_flank</td>\n",
       "      <td>Flank-coding region (Transcript)</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1131</th>\n",
       "      <td>coding_gene_flank</td>\n",
       "      <td>Flank-coding region (Gene)</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1132</th>\n",
       "      <td>5utr</td>\n",
       "      <td>5' UTR</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1133</th>\n",
       "      <td>3utr</td>\n",
       "      <td>3' UTR</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1134</th>\n",
       "      <td>gene_exon</td>\n",
       "      <td>Exon sequences</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1135</th>\n",
       "      <td>cdna</td>\n",
       "      <td>cDNA sequences</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1136</th>\n",
       "      <td>coding</td>\n",
       "      <td>Coding sequence</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1137</th>\n",
       "      <td>peptide</td>\n",
       "      <td>Peptide</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1138</th>\n",
       "      <td>upstream_flank</td>\n",
       "      <td>upstream_flank</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1139</th>\n",
       "      <td>downstream_flank</td>\n",
       "      <td>downstream_flank</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1140</th>\n",
       "      <td>ensembl_gene_id</td>\n",
       "      <td>Gene stable ID</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1141</th>\n",
       "      <td>description</td>\n",
       "      <td>Gene description</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1142</th>\n",
       "      <td>external_gene_name</td>\n",
       "      <td>Gene name</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1143</th>\n",
       "      <td>external_gene_source</td>\n",
       "      <td>Source of gene name</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1144</th>\n",
       "      <td>chromosome_name</td>\n",
       "      <td>Chromosome/scaffold name</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1145</th>\n",
       "      <td>start_position</td>\n",
       "      <td>Gene start (bp)</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1146</th>\n",
       "      <td>end_position</td>\n",
       "      <td>Gene end (bp)</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1147</th>\n",
       "      <td>gene_biotype</td>\n",
       "      <td>Gene type</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1148</th>\n",
       "      <td>uniparc</td>\n",
       "      <td>UniParc ID</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1149</th>\n",
       "      <td>uniprotswissprot</td>\n",
       "      <td>UniProtKB/Swiss-Prot ID</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1150</th>\n",
       "      <td>uniprotsptrembl</td>\n",
       "      <td>UniProtKB/TrEMBL ID</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1151</th>\n",
       "      <td>cdna_coding_start</td>\n",
       "      <td>CDS start (within cDNA)</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1152</th>\n",
       "      <td>cdna_coding_end</td>\n",
       "      <td>CDS end (within cDNA)</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1153</th>\n",
       "      <td>5_utr_start</td>\n",
       "      <td>5' UTR start</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1154</th>\n",
       "      <td>5_utr_end</td>\n",
       "      <td>5' UTR end</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1155</th>\n",
       "      <td>3_utr_start</td>\n",
       "      <td>3' UTR start</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1156</th>\n",
       "      <td>3_utr_end</td>\n",
       "      <td>3' UTR end</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1157</th>\n",
       "      <td>ensembl_transcript_id</td>\n",
       "      <td>Transcript stable ID</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1158</th>\n",
       "      <td>ensembl_peptide_id</td>\n",
       "      <td>Protein stable ID</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1159</th>\n",
       "      <td>transcript_biotype</td>\n",
       "      <td>Transcript type</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1160</th>\n",
       "      <td>strand</td>\n",
       "      <td>Strand</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1161</th>\n",
       "      <td>transcript_start</td>\n",
       "      <td>Transcript start (bp)</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1162</th>\n",
       "      <td>transcript_end</td>\n",
       "      <td>Transcript end (bp)</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1163</th>\n",
       "      <td>transcription_start_site</td>\n",
       "      <td>Transcription start site (TSS)</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1164</th>\n",
       "      <td>transcript_length</td>\n",
       "      <td>Transcript length (including UTRs and CDS)</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1165</th>\n",
       "      <td>cds_length</td>\n",
       "      <td>CDS Length</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1166</th>\n",
       "      <td>cds_start</td>\n",
       "      <td>CDS start</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1167</th>\n",
       "      <td>cds_end</td>\n",
       "      <td>CDS end</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1168</th>\n",
       "      <td>ensembl_exon_id</td>\n",
       "      <td>Exon stable ID</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1169</th>\n",
       "      <td>exon_chrom_start</td>\n",
       "      <td>Exon region start (bp)</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1170</th>\n",
       "      <td>exon_chrom_end</td>\n",
       "      <td>Exon region end (bp)</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1171</th>\n",
       "      <td>strand</td>\n",
       "      <td>Strand</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1172</th>\n",
       "      <td>rank</td>\n",
       "      <td>Exon rank in transcript</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1173</th>\n",
       "      <td>phase</td>\n",
       "      <td>Start phase</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1174</th>\n",
       "      <td>end_phase</td>\n",
       "      <td>End phase</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1175</th>\n",
       "      <td>cdna_coding_start</td>\n",
       "      <td>cDNA coding start</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1176</th>\n",
       "      <td>cdna_coding_end</td>\n",
       "      <td>cDNA coding end</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1177</th>\n",
       "      <td>genomic_coding_start</td>\n",
       "      <td>Genomic coding start</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1178</th>\n",
       "      <td>genomic_coding_end</td>\n",
       "      <td>Genomic coding end</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1179</th>\n",
       "      <td>is_constitutive</td>\n",
       "      <td>Constitutive exon</td>\n",
       "      <td>sequences</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1180 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               name  \\\n",
       "0                                   ensembl_gene_id   \n",
       "1                             ensembl_transcript_id   \n",
       "2                                ensembl_peptide_id   \n",
       "3                                   ensembl_exon_id   \n",
       "4                                       description   \n",
       "5                                   chromosome_name   \n",
       "6                                    start_position   \n",
       "7                                      end_position   \n",
       "8                                            strand   \n",
       "9                                              band   \n",
       "10                                 transcript_start   \n",
       "11                                   transcript_end   \n",
       "12                         transcription_start_site   \n",
       "13                                transcript_length   \n",
       "14                               external_gene_name   \n",
       "15                             external_gene_source   \n",
       "16                                 transcript_count   \n",
       "17                       percentage_gene_gc_content   \n",
       "18                                     gene_biotype   \n",
       "19                               transcript_biotype   \n",
       "20                                           source   \n",
       "21                                transcript_source   \n",
       "22                                            go_id   \n",
       "23                                        name_1006   \n",
       "24                                  definition_1006   \n",
       "25                                  go_linkage_type   \n",
       "26                                   namespace_1003   \n",
       "27                             goslim_goa_accession   \n",
       "28                           goslim_goa_description   \n",
       "29                                           chembl   \n",
       "30                                             embl   \n",
       "31                                       protein_id   \n",
       "32                                      kegg_enzyme   \n",
       "33                                         maizegdb   \n",
       "34                                           merops   \n",
       "35                                mirbase_accession   \n",
       "36                                       mirbase_id   \n",
       "37                                       entrezgene   \n",
       "38                                              pdb   \n",
       "39                           plant_reactome_pathway   \n",
       "40                          plant_reactome_reaction   \n",
       "41                                       refseq_dna   \n",
       "42                                   refseq_peptide   \n",
       "43                                             rfam   \n",
       "44                                           string   \n",
       "45                                      trnascan_se   \n",
       "46                                          uniparc   \n",
       "47                                  uniprotvarsplic   \n",
       "48                                 uniprotswissprot   \n",
       "49                                  uniprotsptrembl   \n",
       "50                                    affy_gpl16720   \n",
       "51                                       affy_maize   \n",
       "52                            agilent_g2519f_016047   \n",
       "53                                              cdd   \n",
       "54                                        cdd_start   \n",
       "55                                          cdd_end   \n",
       "56                                           gene3d   \n",
       "57                                     gene3d_start   \n",
       "58                                       gene3d_end   \n",
       "59                                            hamap   \n",
       "60                                      hamap_start   \n",
       "61                                        hamap_end   \n",
       "62                                       hmmpanther   \n",
       "63                                 hmmpanther_start   \n",
       "64                                   hmmpanther_end   \n",
       "65                                             pfam   \n",
       "66                                       pfam_start   \n",
       "67                                         pfam_end   \n",
       "68                                            pirsf   \n",
       "69                                      pirsf_start   \n",
       "70                                        pirsf_end   \n",
       "71                                           prints   \n",
       "72                                     prints_start   \n",
       "73                                       prints_end   \n",
       "74                                      blastprodom   \n",
       "75                                blastprodom_start   \n",
       "76                                  blastprodom_end   \n",
       "77                                      scanprosite   \n",
       "78                                scanprosite_start   \n",
       "79                                  scanprosite_end   \n",
       "80                                           pfscan   \n",
       "81                                     pfscan_start   \n",
       "82                                       pfscan_end   \n",
       "83                                             sfld   \n",
       "84                                       sfld_start   \n",
       "85                                         sfld_end   \n",
       "86                                            smart   \n",
       "87                                      smart_start   \n",
       "88                                        smart_end   \n",
       "89                                      superfamily   \n",
       "90                                superfamily_start   \n",
       "91                                  superfamily_end   \n",
       "92                                          tigrfam   \n",
       "93                                    tigrfam_start   \n",
       "94                                      tigrfam_end   \n",
       "95                                         interpro   \n",
       "96                       interpro_short_description   \n",
       "97                             interpro_description   \n",
       "98                                   interpro_start   \n",
       "99                                     interpro_end   \n",
       "...                                             ...   \n",
       "1080  zmays_eg_paralog_canonical_transcript_protein   \n",
       "1081                       zmays_eg_paralog_subtype   \n",
       "1082                zmays_eg_paralog_orthology_type   \n",
       "1083                       zmays_eg_paralog_perc_id   \n",
       "1084                    zmays_eg_paralog_perc_id_r1   \n",
       "1085                            zmays_eg_paralog_dn   \n",
       "1086                            zmays_eg_paralog_ds   \n",
       "1087           zmays_eg_paralog_paralogy_confidence   \n",
       "1088                                ensembl_gene_id   \n",
       "1089                          ensembl_transcript_id   \n",
       "1090                             ensembl_peptide_id   \n",
       "1091                                chromosome_name   \n",
       "1092                                 start_position   \n",
       "1093                                   end_position   \n",
       "1094                                         strand   \n",
       "1095                                           band   \n",
       "1096                             external_gene_name   \n",
       "1097                           external_gene_source   \n",
       "1098                               transcript_count   \n",
       "1099                     percentage_gene_gc_content   \n",
       "1100                                    description   \n",
       "1101                                 variation_name   \n",
       "1102                     germ_line_variation_source   \n",
       "1103                             source_description   \n",
       "1104                                         allele   \n",
       "1105                                      validated   \n",
       "1106                                      mapweight   \n",
       "1107                                   minor_allele   \n",
       "1108                              minor_allele_freq   \n",
       "1109                             minor_allele_count   \n",
       "1110                          clinical_significance   \n",
       "1111                            transcript_location   \n",
       "1112                          snp_chromosome_strand   \n",
       "1113                               peptide_location   \n",
       "1114                               chromosome_start   \n",
       "1115                                 chromosome_end   \n",
       "1116                       polyphen_prediction_2076   \n",
       "1117                            polyphen_score_2076   \n",
       "1118                           sift_prediction_2076   \n",
       "1119                                sift_score_2076   \n",
       "1120                    distance_to_transcript_2076   \n",
       "1121                                 cds_start_2076   \n",
       "1122                                   cds_end_2076   \n",
       "1123                                  peptide_shift   \n",
       "1124                              synonymous_status   \n",
       "1125                             allele_string_2076   \n",
       "1126                         transcript_exon_intron   \n",
       "1127                               gene_exon_intron   \n",
       "1128                               transcript_flank   \n",
       "1129                                     gene_flank   \n",
       "1130                        coding_transcript_flank   \n",
       "1131                              coding_gene_flank   \n",
       "1132                                           5utr   \n",
       "1133                                           3utr   \n",
       "1134                                      gene_exon   \n",
       "1135                                           cdna   \n",
       "1136                                         coding   \n",
       "1137                                        peptide   \n",
       "1138                                 upstream_flank   \n",
       "1139                               downstream_flank   \n",
       "1140                                ensembl_gene_id   \n",
       "1141                                    description   \n",
       "1142                             external_gene_name   \n",
       "1143                           external_gene_source   \n",
       "1144                                chromosome_name   \n",
       "1145                                 start_position   \n",
       "1146                                   end_position   \n",
       "1147                                   gene_biotype   \n",
       "1148                                        uniparc   \n",
       "1149                               uniprotswissprot   \n",
       "1150                                uniprotsptrembl   \n",
       "1151                              cdna_coding_start   \n",
       "1152                                cdna_coding_end   \n",
       "1153                                    5_utr_start   \n",
       "1154                                      5_utr_end   \n",
       "1155                                    3_utr_start   \n",
       "1156                                      3_utr_end   \n",
       "1157                          ensembl_transcript_id   \n",
       "1158                             ensembl_peptide_id   \n",
       "1159                             transcript_biotype   \n",
       "1160                                         strand   \n",
       "1161                               transcript_start   \n",
       "1162                                 transcript_end   \n",
       "1163                       transcription_start_site   \n",
       "1164                              transcript_length   \n",
       "1165                                     cds_length   \n",
       "1166                                      cds_start   \n",
       "1167                                        cds_end   \n",
       "1168                                ensembl_exon_id   \n",
       "1169                               exon_chrom_start   \n",
       "1170                                 exon_chrom_end   \n",
       "1171                                         strand   \n",
       "1172                                           rank   \n",
       "1173                                          phase   \n",
       "1174                                      end_phase   \n",
       "1175                              cdna_coding_start   \n",
       "1176                                cdna_coding_end   \n",
       "1177                           genomic_coding_start   \n",
       "1178                             genomic_coding_end   \n",
       "1179                                is_constitutive   \n",
       "\n",
       "                                            description          page  \n",
       "0                                        Gene stable ID  feature_page  \n",
       "1                                  Transcript stable ID  feature_page  \n",
       "2                                     Protein stable ID  feature_page  \n",
       "3                                        Exon stable ID  feature_page  \n",
       "4                                      Gene description  feature_page  \n",
       "5                              Chromosome/scaffold name  feature_page  \n",
       "6                                       Gene start (bp)  feature_page  \n",
       "7                                         Gene end (bp)  feature_page  \n",
       "8                                                Strand  feature_page  \n",
       "9                                        Karyotype band  feature_page  \n",
       "10                                Transcript start (bp)  feature_page  \n",
       "11                                  Transcript end (bp)  feature_page  \n",
       "12                       Transcription start site (TSS)  feature_page  \n",
       "13           Transcript length (including UTRs and CDS)  feature_page  \n",
       "14                                            Gene name  feature_page  \n",
       "15                                  Source of gene name  feature_page  \n",
       "16                                     Transcript count  feature_page  \n",
       "17                                    Gene % GC content  feature_page  \n",
       "18                                            Gene type  feature_page  \n",
       "19                                      Transcript type  feature_page  \n",
       "20                                        Source (gene)  feature_page  \n",
       "21                                  Source (transcript)  feature_page  \n",
       "22                                    GO term accession  feature_page  \n",
       "23                                         GO term name  feature_page  \n",
       "24                                   GO term definition  feature_page  \n",
       "25                                GO term evidence code  feature_page  \n",
       "26                                            GO domain  feature_page  \n",
       "27                              GOSlim GOA Accession(s)  feature_page  \n",
       "28                               GOSlim GOA Description  feature_page  \n",
       "29                                            ChEMBL ID  feature_page  \n",
       "30                       European Nucleotide Archive ID  feature_page  \n",
       "31                                     INSDC protein ID  feature_page  \n",
       "32                           KEGG Pathway and Enzyme ID  feature_page  \n",
       "33                                          MaizeGDB ID  feature_page  \n",
       "34                   MEROPS - the Peptidase Database ID  feature_page  \n",
       "35                                    miRBase accession  feature_page  \n",
       "36                                           miRBase ID  feature_page  \n",
       "37                                         NCBI gene ID  feature_page  \n",
       "38                                               PDB ID  feature_page  \n",
       "39                            Plant Reactome Pathway ID  feature_page  \n",
       "40                           Plant Reactome Reaction ID  feature_page  \n",
       "41                                        RefSeq DNA ID  feature_page  \n",
       "42                                    RefSeq peptide ID  feature_page  \n",
       "43                                              RFAM ID  feature_page  \n",
       "44                                            STRING ID  feature_page  \n",
       "45                                       tRNAScan-SE ID  feature_page  \n",
       "46                                           UniParc ID  feature_page  \n",
       "47                           UniProtKB/SpliceVariant ID  feature_page  \n",
       "48                              UniProtKB/Swiss-Prot ID  feature_page  \n",
       "49                                  UniProtKB/TrEMBL ID  feature_page  \n",
       "50                                  AFFY GPL16720 probe  feature_page  \n",
       "51                                     AFFY maize probe  feature_page  \n",
       "52                          AGILENT G2519F 016047 probe  feature_page  \n",
       "53                                               CDD ID  feature_page  \n",
       "54                                            CDD start  feature_page  \n",
       "55                                              CDD end  feature_page  \n",
       "56                                            Gene3D ID  feature_page  \n",
       "57                                         Gene3D start  feature_page  \n",
       "58                                           Gene3D end  feature_page  \n",
       "59                                             HAMAP ID  feature_page  \n",
       "60                                          HAMAP start  feature_page  \n",
       "61                                            HAMAP end  feature_page  \n",
       "62                                           PANTHER ID  feature_page  \n",
       "63                                        PANTHER start  feature_page  \n",
       "64                                          PANTHER end  feature_page  \n",
       "65                                              Pfam ID  feature_page  \n",
       "66                                           Pfam start  feature_page  \n",
       "67                                             Pfam end  feature_page  \n",
       "68                                             PIRSF ID  feature_page  \n",
       "69                                          PIRSF start  feature_page  \n",
       "70                                            PIRSF end  feature_page  \n",
       "71                                            Prints ID  feature_page  \n",
       "72                                         Prints start  feature_page  \n",
       "73                                           Prints end  feature_page  \n",
       "74                                            ProDom ID  feature_page  \n",
       "75                                         ProDom start  feature_page  \n",
       "76                                           ProDom end  feature_page  \n",
       "77                                  PROSITE patterns ID  feature_page  \n",
       "78                               PROSITE patterns start  feature_page  \n",
       "79                                 PROSITE patterns end  feature_page  \n",
       "80                                  PROSITE profiles ID  feature_page  \n",
       "81                               PROSITE profiles start  feature_page  \n",
       "82                                 PROSITE profiles end  feature_page  \n",
       "83                                              SFLD ID  feature_page  \n",
       "84                                           SFLD start  feature_page  \n",
       "85                                             SFLD end  feature_page  \n",
       "86                                             SMART ID  feature_page  \n",
       "87                                          SMART start  feature_page  \n",
       "88                                            SMART end  feature_page  \n",
       "89                                       Superfamily ID  feature_page  \n",
       "90                                    Superfamily start  feature_page  \n",
       "91                                      Superfamily end  feature_page  \n",
       "92                                           TIGRFAM ID  feature_page  \n",
       "93                                        TIGRFAM start  feature_page  \n",
       "94                                          TIGRFAM end  feature_page  \n",
       "95                                          Interpro ID  feature_page  \n",
       "96                           Interpro Short Description  feature_page  \n",
       "97                                 Interpro Description  feature_page  \n",
       "98                                       Interpro start  feature_page  \n",
       "99                                         Interpro end  feature_page  \n",
       "...                                                 ...           ...  \n",
       "1080           Paralogue query protein or transcript ID      homologs  \n",
       "1081       Paralogue last common ancestor with Zea mays      homologs  \n",
       "1082                   Zea mays paralogue homology type      homologs  \n",
       "1083  Paralogue %id. target Zea mays gene identical ...      homologs  \n",
       "1084  Paralogue %id. query gene identical to target ...      homologs  \n",
       "1085                         Paralogue dN with Zea mays      homologs  \n",
       "1086                         Paralogue dS with Zea mays      homologs  \n",
       "1087       Zea mays paralogy confidence [0 low, 1 high]      homologs  \n",
       "1088                                     Gene stable ID           snp  \n",
       "1089                               Transcript stable ID           snp  \n",
       "1090                                  Protein stable ID           snp  \n",
       "1091                           Chromosome/scaffold name           snp  \n",
       "1092                                    Gene start (bp)           snp  \n",
       "1093                                      Gene end (bp)           snp  \n",
       "1094                                             Strand           snp  \n",
       "1095                                     Karyotype band           snp  \n",
       "1096                                          Gene name           snp  \n",
       "1097                                Source of gene name           snp  \n",
       "1098                                   Transcript count           snp  \n",
       "1099                                  Gene % GC content           snp  \n",
       "1100                                   Gene description           snp  \n",
       "1101                                       Variant name           snp  \n",
       "1102                                     Variant source           snp  \n",
       "1103                         Variant source description           snp  \n",
       "1104                                    Variant alleles           snp  \n",
       "1105                        Variant supporting evidence           snp  \n",
       "1106                                          Mapweight           snp  \n",
       "1107                                       Minor allele           snp  \n",
       "1108                             Minor allele frequency           snp  \n",
       "1109                                 Minor allele count           snp  \n",
       "1110                              Clinical significance           snp  \n",
       "1111                           Transcript location (bp)           snp  \n",
       "1112                          Variant chromosome Strand           snp  \n",
       "1113                              Protein location (aa)           snp  \n",
       "1114            chromosome/scaffold position start (bp)           snp  \n",
       "1115              Chromosome/scaffold position end (bp)           snp  \n",
       "1116                                PolyPhen prediction           snp  \n",
       "1117                                     PolyPhen score           snp  \n",
       "1118                                    SIFT prediction           snp  \n",
       "1119                                         SIFT score           snp  \n",
       "1120                             Distance to transcript           snp  \n",
       "1121                                          CDS start           snp  \n",
       "1122                                            CDS end           snp  \n",
       "1123                                     Protein allele           snp  \n",
       "1124                                Variant consequence           snp  \n",
       "1125                        Consequence specific allele           snp  \n",
       "1126                             Unspliced (Transcript)     sequences  \n",
       "1127                                   Unspliced (Gene)     sequences  \n",
       "1128                                 Flank (Transcript)     sequences  \n",
       "1129                                       Flank (Gene)     sequences  \n",
       "1130                   Flank-coding region (Transcript)     sequences  \n",
       "1131                         Flank-coding region (Gene)     sequences  \n",
       "1132                                             5' UTR     sequences  \n",
       "1133                                             3' UTR     sequences  \n",
       "1134                                     Exon sequences     sequences  \n",
       "1135                                     cDNA sequences     sequences  \n",
       "1136                                    Coding sequence     sequences  \n",
       "1137                                            Peptide     sequences  \n",
       "1138                                     upstream_flank     sequences  \n",
       "1139                                   downstream_flank     sequences  \n",
       "1140                                     Gene stable ID     sequences  \n",
       "1141                                   Gene description     sequences  \n",
       "1142                                          Gene name     sequences  \n",
       "1143                                Source of gene name     sequences  \n",
       "1144                           Chromosome/scaffold name     sequences  \n",
       "1145                                    Gene start (bp)     sequences  \n",
       "1146                                      Gene end (bp)     sequences  \n",
       "1147                                          Gene type     sequences  \n",
       "1148                                         UniParc ID     sequences  \n",
       "1149                            UniProtKB/Swiss-Prot ID     sequences  \n",
       "1150                                UniProtKB/TrEMBL ID     sequences  \n",
       "1151                            CDS start (within cDNA)     sequences  \n",
       "1152                              CDS end (within cDNA)     sequences  \n",
       "1153                                       5' UTR start     sequences  \n",
       "1154                                         5' UTR end     sequences  \n",
       "1155                                       3' UTR start     sequences  \n",
       "1156                                         3' UTR end     sequences  \n",
       "1157                               Transcript stable ID     sequences  \n",
       "1158                                  Protein stable ID     sequences  \n",
       "1159                                    Transcript type     sequences  \n",
       "1160                                             Strand     sequences  \n",
       "1161                              Transcript start (bp)     sequences  \n",
       "1162                                Transcript end (bp)     sequences  \n",
       "1163                     Transcription start site (TSS)     sequences  \n",
       "1164         Transcript length (including UTRs and CDS)     sequences  \n",
       "1165                                         CDS Length     sequences  \n",
       "1166                                          CDS start     sequences  \n",
       "1167                                            CDS end     sequences  \n",
       "1168                                     Exon stable ID     sequences  \n",
       "1169                             Exon region start (bp)     sequences  \n",
       "1170                               Exon region end (bp)     sequences  \n",
       "1171                                             Strand     sequences  \n",
       "1172                            Exon rank in transcript     sequences  \n",
       "1173                                        Start phase     sequences  \n",
       "1174                                          End phase     sequences  \n",
       "1175                                  cDNA coding start     sequences  \n",
       "1176                                    cDNA coding end     sequences  \n",
       "1177                               Genomic coding start     sequences  \n",
       "1178                                 Genomic coding end     sequences  \n",
       "1179                                  Constitutive exon     sequences  \n",
       "\n",
       "[1180 rows x 3 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "pd.set_option('display.max_rows', 200)\n",
    "a =%R salida\n",
    "display(a)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "especie = \"Zea mays\"\n",
    "especie = paste(\"^\", especie, sep=\"\")\n",
    "\n",
    "library(biomaRt)\n",
    "library(stringr)\n",
    "\n",
    "ensembl=useMart(\"plants_mart\", host=\"plants.ensembl.org\")\n",
    "lista_datasets = listDatasets(ensembl)\n",
    "dataset = lista_datasets[which(str_detect(lista_datasets$description, especie)), 1]\n",
    "ensembl = useDataset(dataset,mart=ensembl)\n",
    "salida = 0\n",
    "tryCatch({\n",
    "    genes = getBM(attributes=c('ensembl_gene_id'), filters = 'biotype', values = 'lincRNA', mart = ensembl)\n",
    "    salida = nrow(genes)\n",
    "}, error = function(err) {\n",
    "    salida = 0\n",
    "})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2532]\n"
     ]
    }
   ],
   "source": [
    "a =%R salida\n",
    "print(a)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Estamos listos para realizar el conteo. Se tiene en la base de datos local una tabla conteo_lncrna_especie, que posee la lista de especies de referencia para el maíz, con una columna de conteo, donde insertaremos el resultado de las consultas. El código será enteramente en R."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "library(RMySQL)\n",
    "\n",
    "dsn_database = \"tesis2\"\n",
    "dsn_hostname = \"localhost\"\n",
    "dsn_port = 3306\n",
    "dsn_uid = \"root\"\n",
    "dsn_pwd = \"\"\n",
    "\n",
    "conn = dbConnect(MySQL(), user=dsn_uid, password=dsn_pwd, dbname=dsn_database, host=dsn_hostname)\n",
    "rs = dbSendQuery(conn, \"select especie from conteo_lncrna_especie where para_especie like 'Zea mays%'\")\n",
    "especies = dbFetch(rs)\n",
    "dbClearResult(rs)\n",
    "\n",
    "library(biomaRt)\n",
    "library(stringr)\n",
    "ensembl_mart=useMart(\"plants_mart\", host=\"plants.ensembl.org\")\n",
    "lista_datasets = listDatasets(ensembl)\n",
    "\n",
    "for(i in 1:nrow(especies)) {\n",
    "    la_especie = especies[i,]\n",
    "    especie = paste(\"^\", la_especie, sep=\"\")\n",
    "    \n",
    "    conteo = 0\n",
    "    tryCatch({\n",
    "        dataset = lista_datasets[which(str_detect(lista_datasets$description, especie)), 1]\n",
    "        ensembl = useDataset(dataset,mart=ensembl_mart)\n",
    "        genes = getBM(attributes=c('ensembl_gene_id'), filters = 'biotype', values = 'protein_coding', mart = ensembl)\n",
    "        conteo = nrow(genes)\n",
    "    }, error = function(err) {\n",
    "        conteo = 0\n",
    "    })\n",
    "    query = sprintf(\"update conteo_lncrna_especie set conteo_pct = %d where especie='%s'\", conteo, la_especie)\n",
    "    dbSendQuery(conn, query)\n",
    "}\n",
    "\n",
    "dbDisconnect(conn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>para_especie</th>\n",
       "      <th>especie</th>\n",
       "      <th>conteo_ncRNA</th>\n",
       "      <th>conteo_lncRNA</th>\n",
       "      <th>conteo_lincRNA</th>\n",
       "      <th>conteo_pct</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Brassica napus</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>101040</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Zea mays</td>\n",
       "      <td>Triticum dicoccoides</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>62569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Zea mays</td>\n",
       "      <td>Triticum dicoccoides</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>62569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Zea mays</td>\n",
       "      <td>Triticum dicoccoides</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>62569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Brassica oleracea</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>59220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Helianthus annuus</td>\n",
       "      <td>4659</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>52191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Zea mays</td>\n",
       "      <td>Oryza sativa Indica Group</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>40745</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Zea mays</td>\n",
       "      <td>Aegilops tauschii</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>39614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Solanum tuberosum</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>39021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Gossypium raimondii</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38208</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Zea mays</td>\n",
       "      <td>Oryza rufipogon</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>37071</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Zea mays</td>\n",
       "      <td>Oryza nivara</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>36313</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Zea mays</td>\n",
       "      <td>Oryza sativa Japonica Group</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>35821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Zea mays</td>\n",
       "      <td>Oryza glumipatula</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>35735</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Zea mays</td>\n",
       "      <td>Oryza barthii</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>34575</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Zea mays</td>\n",
       "      <td>Brachypodium distachyon</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>34310</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Zea mays</td>\n",
       "      <td>Sorghum bicolor</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>34118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Vigna angularis</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>33860</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Zea mays</td>\n",
       "      <td>Triticum urartu</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>33483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Zea mays</td>\n",
       "      <td>Oryza glaberrima</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>33164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Zea mays</td>\n",
       "      <td>Oryza brachyantha</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>32037</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Zea mays</td>\n",
       "      <td>Oryza punctata</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>31762</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Zea mays</td>\n",
       "      <td>Oryza longistaminata</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>31686</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Zea mays</td>\n",
       "      <td>Oryza meridionalis</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>29308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Theobroma cacao</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>29188</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Zea mays</td>\n",
       "      <td>Leersia perrieri</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>29078</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Cucumis sativus</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>23780</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Chlamydomonas reinhardtii</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>17743</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Chondrus crispus</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9807</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Galdieria sulphuraria</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6622</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Cyanidioschyzon merolae</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4998</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>Zea mays</td>\n",
       "      <td>Triticum aestivum</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>Zea mays</td>\n",
       "      <td>Triticum aestivum</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>Zea mays</td>\n",
       "      <td>Triticum aestivum</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>Zea mays</td>\n",
       "      <td>Hordeum vulgare</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Arabidopsis halleri</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Arabidopsis thaliana</td>\n",
       "      <td>299</td>\n",
       "      <td>3481</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Nicotiana attenuata</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Brassica rapa</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Solanum lycopersicum</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Corchorus capsularis</td>\n",
       "      <td>667</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Daucus carota</td>\n",
       "      <td>780</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Manihot esculenta</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Populus trichocarpa</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Medicago truncatula</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Trifolium pratense</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Dioscorea rotundata</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Vigna radiata</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>Zea mays</td>\n",
       "      <td>Setaria italica</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Glycine max</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Lupinus angustifolius</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>Zea mays</td>\n",
       "      <td>Musa acuminata</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Prunus persica</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Vitis vinifera</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Beta vulgaris</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Amborella trichopoda</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Selaginella moellendorffii</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Physcomitrella patens</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Ostreococcus lucimarinus</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Arabidopsis lyrata</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Caenorhabditis elegans</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Fruitfly</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>C.savignyi</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Human</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Saccharomyces cerevisiae</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>Zea mays</td>\n",
       "      <td>Triticum aestivum</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>Zea mays</td>\n",
       "      <td>Triticum aestivum</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>Zea mays2</td>\n",
       "      <td>Phaseolus vulgaris</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   para_especie                      especie  conteo_ncRNA  conteo_lncRNA  \\\n",
       "0     Zea mays2               Brassica napus             0              0   \n",
       "1      Zea mays         Triticum dicoccoides             0              0   \n",
       "2      Zea mays         Triticum dicoccoides             0              0   \n",
       "3      Zea mays         Triticum dicoccoides             0              0   \n",
       "4     Zea mays2            Brassica oleracea             0              0   \n",
       "5     Zea mays2            Helianthus annuus          4659              0   \n",
       "6      Zea mays    Oryza sativa Indica Group             0              0   \n",
       "7      Zea mays            Aegilops tauschii             0              0   \n",
       "8     Zea mays2            Solanum tuberosum             0              0   \n",
       "9     Zea mays2          Gossypium raimondii             2              0   \n",
       "10     Zea mays              Oryza rufipogon             0              0   \n",
       "11     Zea mays                 Oryza nivara             0              0   \n",
       "12     Zea mays  Oryza sativa Japonica Group             0              0   \n",
       "13     Zea mays            Oryza glumipatula             0              0   \n",
       "14     Zea mays                Oryza barthii             2              0   \n",
       "15     Zea mays      Brachypodium distachyon             0              0   \n",
       "16     Zea mays              Sorghum bicolor             0              0   \n",
       "17    Zea mays2              Vigna angularis             0              0   \n",
       "18     Zea mays              Triticum urartu             0              0   \n",
       "19     Zea mays             Oryza glaberrima             0              0   \n",
       "20     Zea mays            Oryza brachyantha             0              0   \n",
       "21     Zea mays               Oryza punctata             0              0   \n",
       "22     Zea mays         Oryza longistaminata             3              0   \n",
       "23     Zea mays           Oryza meridionalis             2              0   \n",
       "24    Zea mays2              Theobroma cacao             0              0   \n",
       "25     Zea mays             Leersia perrieri             2              0   \n",
       "26    Zea mays2              Cucumis sativus             0              0   \n",
       "27    Zea mays2    Chlamydomonas reinhardtii             0              0   \n",
       "28    Zea mays2             Chondrus crispus             0              0   \n",
       "29    Zea mays2        Galdieria sulphuraria             0              0   \n",
       "30    Zea mays2      Cyanidioschyzon merolae             0              0   \n",
       "31     Zea mays            Triticum aestivum             0              0   \n",
       "32     Zea mays            Triticum aestivum             0              0   \n",
       "33     Zea mays            Triticum aestivum             0              0   \n",
       "34     Zea mays              Hordeum vulgare             0              0   \n",
       "35    Zea mays2          Arabidopsis halleri             0              0   \n",
       "36    Zea mays2         Arabidopsis thaliana           299           3481   \n",
       "37    Zea mays2          Nicotiana attenuata             0              0   \n",
       "38    Zea mays2                Brassica rapa             0              0   \n",
       "39    Zea mays2         Solanum lycopersicum             1              0   \n",
       "40    Zea mays2         Corchorus capsularis           667              0   \n",
       "41    Zea mays2                Daucus carota           780              0   \n",
       "42    Zea mays2            Manihot esculenta             3              0   \n",
       "43    Zea mays2          Populus trichocarpa             0              0   \n",
       "44    Zea mays2          Medicago truncatula             1              0   \n",
       "45    Zea mays2           Trifolium pratense             0              0   \n",
       "46    Zea mays2          Dioscorea rotundata             0              0   \n",
       "47    Zea mays2                Vigna radiata             0              0   \n",
       "48     Zea mays              Setaria italica             0              0   \n",
       "49    Zea mays2                  Glycine max             2              0   \n",
       "50    Zea mays2        Lupinus angustifolius             0              0   \n",
       "51     Zea mays               Musa acuminata             0              0   \n",
       "52    Zea mays2               Prunus persica             1              0   \n",
       "53    Zea mays2               Vitis vinifera             0              0   \n",
       "54    Zea mays2                Beta vulgaris             0              0   \n",
       "55    Zea mays2         Amborella trichopoda             0              0   \n",
       "56    Zea mays2   Selaginella moellendorffii             0              0   \n",
       "57    Zea mays2        Physcomitrella patens             0              0   \n",
       "58    Zea mays2     Ostreococcus lucimarinus             0              0   \n",
       "59    Zea mays2           Arabidopsis lyrata             0              0   \n",
       "60    Zea mays2       Caenorhabditis elegans             0              0   \n",
       "61    Zea mays2                     Fruitfly             0              0   \n",
       "62    Zea mays2                   C.savignyi             0              0   \n",
       "63    Zea mays2                        Human             0              0   \n",
       "64    Zea mays2     Saccharomyces cerevisiae             0              0   \n",
       "65     Zea mays            Triticum aestivum             0              0   \n",
       "66     Zea mays            Triticum aestivum             0              0   \n",
       "67    Zea mays2           Phaseolus vulgaris             0              0   \n",
       "\n",
       "    conteo_lincRNA  conteo_pct  \n",
       "0                0      101040  \n",
       "1                0       62569  \n",
       "2                0       62569  \n",
       "3                0       62569  \n",
       "4                0       59220  \n",
       "5                0       52191  \n",
       "6                0       40745  \n",
       "7                0       39614  \n",
       "8                0       39021  \n",
       "9                0       38208  \n",
       "10               0       37071  \n",
       "11               0       36313  \n",
       "12               0       35821  \n",
       "13               0       35735  \n",
       "14               0       34575  \n",
       "15               0       34310  \n",
       "16               0       34118  \n",
       "17               0       33860  \n",
       "18               0       33483  \n",
       "19               0       33164  \n",
       "20               0       32037  \n",
       "21               0       31762  \n",
       "22               0       31686  \n",
       "23               0       29308  \n",
       "24               0       29188  \n",
       "25               0       29078  \n",
       "26               0       23780  \n",
       "27               0       17743  \n",
       "28               0        9807  \n",
       "29               0        6622  \n",
       "30               0        4998  \n",
       "31               0           0  \n",
       "32               0           0  \n",
       "33               0           0  \n",
       "34               0           0  \n",
       "35               0           0  \n",
       "36               0           0  \n",
       "37               0           0  \n",
       "38               0           0  \n",
       "39               0           0  \n",
       "40               0           0  \n",
       "41               0           0  \n",
       "42               0           0  \n",
       "43               0           0  \n",
       "44               0           0  \n",
       "45               0           0  \n",
       "46               0           0  \n",
       "47               0           0  \n",
       "48               0           0  \n",
       "49               0           0  \n",
       "50               0           0  \n",
       "51               0           0  \n",
       "52               0           0  \n",
       "53               0           0  \n",
       "54               0           0  \n",
       "55               0           0  \n",
       "56               0           0  \n",
       "57               0           0  \n",
       "58               0           0  \n",
       "59               0           0  \n",
       "60               0           0  \n",
       "61               0           0  \n",
       "62               0           0  \n",
       "63               0           0  \n",
       "64               0           0  \n",
       "65               0           0  \n",
       "66               0           0  \n",
       "67               0           0  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import mysql.connector\n",
    "import pandas as pd\n",
    "\n",
    "conn = mysql.connector.connect(\n",
    "   host=\"localhost\",\n",
    "   database=\"tesis2\",\n",
    "   user=\"root\",\n",
    "   passwd=\"\"\n",
    ")\n",
    "\n",
    "df = pd.read_sql_query(\"select * from conteo_lncrna_especie order by 6 desc\", conn)\n",
    "display(df)\n",
    "\n",
    "conn.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Tras ejecutar las consultas, descubrimos que no existe ningún lincRNA, apenas una especie con lncRNA, y 4 especies con ncRNA.\n",
    "Será mejor obtener la cantidad de lncRNA en cantatadb. Para ello, ya se descargaron los archivos fasta de transcriptomas de la web de cantatadb: http://cantata.amu.edu.pl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>para_especie</th>\n",
       "      <th>especie</th>\n",
       "      <th>conteo_ncRNA</th>\n",
       "      <th>conteo_lncRNA</th>\n",
       "      <th>conteo_lincRNA</th>\n",
       "      <th>conteo_pct</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Chenopodium_quinoa</td>\n",
       "      <td>None</td>\n",
       "      <td>17526</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Brassica_napus</td>\n",
       "      <td>None</td>\n",
       "      <td>12010</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Malus_domestica</td>\n",
       "      <td>None</td>\n",
       "      <td>10924</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Ananas_comosus</td>\n",
       "      <td>None</td>\n",
       "      <td>10404</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Oryza_rufipogon</td>\n",
       "      <td>None</td>\n",
       "      <td>10261</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Trifolium_pratense</td>\n",
       "      <td>None</td>\n",
       "      <td>10179</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Manihot_esculenta</td>\n",
       "      <td>None</td>\n",
       "      <td>9504</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Oryza_nivara</td>\n",
       "      <td>None</td>\n",
       "      <td>8955</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Brassica_rapa</td>\n",
       "      <td>None</td>\n",
       "      <td>8501</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Hordeum_vulgare</td>\n",
       "      <td>None</td>\n",
       "      <td>7970</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Arabidopsis_lyrata</td>\n",
       "      <td>None</td>\n",
       "      <td>7593</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Cucumis_sativus</td>\n",
       "      <td>None</td>\n",
       "      <td>7348</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Brassica_oleracea</td>\n",
       "      <td>None</td>\n",
       "      <td>7338</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Oryza_barthii</td>\n",
       "      <td>None</td>\n",
       "      <td>7062</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Corchorus_capsularis</td>\n",
       "      <td>None</td>\n",
       "      <td>6459</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Leersia_perrieri</td>\n",
       "      <td>None</td>\n",
       "      <td>6402</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Oryza_brachyantha</td>\n",
       "      <td>None</td>\n",
       "      <td>6004</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Solanum_tuberosum</td>\n",
       "      <td>None</td>\n",
       "      <td>5790</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Amborella_trichopoda</td>\n",
       "      <td>None</td>\n",
       "      <td>5511</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Theobroma_cacao</td>\n",
       "      <td>None</td>\n",
       "      <td>5256</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Brachypodium_distachyon</td>\n",
       "      <td>None</td>\n",
       "      <td>4945</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Solanum_lycopersicum</td>\n",
       "      <td>None</td>\n",
       "      <td>4716</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Vitis_vinifera</td>\n",
       "      <td>None</td>\n",
       "      <td>4542</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Arabidopsis_thaliana</td>\n",
       "      <td>None</td>\n",
       "      <td>4373</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Populus_trichocarpa</td>\n",
       "      <td>None</td>\n",
       "      <td>4322</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Setaria_italica</td>\n",
       "      <td>None</td>\n",
       "      <td>4208</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Medicago_truncatula</td>\n",
       "      <td>None</td>\n",
       "      <td>3590</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Oryza_punctata</td>\n",
       "      <td>None</td>\n",
       "      <td>3459</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Chlamydomonas_reinhardtii</td>\n",
       "      <td>None</td>\n",
       "      <td>3425</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Glycine_max</td>\n",
       "      <td>None</td>\n",
       "      <td>3096</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Musa_acuminata</td>\n",
       "      <td>None</td>\n",
       "      <td>3001</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Prunus_persica</td>\n",
       "      <td>None</td>\n",
       "      <td>2902</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Oryza_sativa_Japonica_Group</td>\n",
       "      <td>None</td>\n",
       "      <td>2788</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Sorghum_bicolor</td>\n",
       "      <td>None</td>\n",
       "      <td>2600</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Selaginella_moellendorffii</td>\n",
       "      <td>None</td>\n",
       "      <td>2267</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Galdieria_sulphuraria</td>\n",
       "      <td>None</td>\n",
       "      <td>1917</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Physcomitrella_patens</td>\n",
       "      <td>None</td>\n",
       "      <td>1498</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>Zea mays cantatadb</td>\n",
       "      <td>Chondrus_crispus</td>\n",
       "      <td>None</td>\n",
       "      <td>224</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          para_especie                      especie conteo_ncRNA  \\\n",
       "0   Zea mays cantatadb           Chenopodium_quinoa         None   \n",
       "1   Zea mays cantatadb               Brassica_napus         None   \n",
       "2   Zea mays cantatadb              Malus_domestica         None   \n",
       "3   Zea mays cantatadb               Ananas_comosus         None   \n",
       "4   Zea mays cantatadb              Oryza_rufipogon         None   \n",
       "5   Zea mays cantatadb           Trifolium_pratense         None   \n",
       "6   Zea mays cantatadb            Manihot_esculenta         None   \n",
       "7   Zea mays cantatadb                 Oryza_nivara         None   \n",
       "8   Zea mays cantatadb                Brassica_rapa         None   \n",
       "9   Zea mays cantatadb              Hordeum_vulgare         None   \n",
       "10  Zea mays cantatadb           Arabidopsis_lyrata         None   \n",
       "11  Zea mays cantatadb              Cucumis_sativus         None   \n",
       "12  Zea mays cantatadb            Brassica_oleracea         None   \n",
       "13  Zea mays cantatadb                Oryza_barthii         None   \n",
       "14  Zea mays cantatadb         Corchorus_capsularis         None   \n",
       "15  Zea mays cantatadb             Leersia_perrieri         None   \n",
       "16  Zea mays cantatadb            Oryza_brachyantha         None   \n",
       "17  Zea mays cantatadb            Solanum_tuberosum         None   \n",
       "18  Zea mays cantatadb         Amborella_trichopoda         None   \n",
       "19  Zea mays cantatadb              Theobroma_cacao         None   \n",
       "20  Zea mays cantatadb      Brachypodium_distachyon         None   \n",
       "21  Zea mays cantatadb         Solanum_lycopersicum         None   \n",
       "22  Zea mays cantatadb               Vitis_vinifera         None   \n",
       "23  Zea mays cantatadb         Arabidopsis_thaliana         None   \n",
       "24  Zea mays cantatadb          Populus_trichocarpa         None   \n",
       "25  Zea mays cantatadb              Setaria_italica         None   \n",
       "26  Zea mays cantatadb          Medicago_truncatula         None   \n",
       "27  Zea mays cantatadb               Oryza_punctata         None   \n",
       "28  Zea mays cantatadb    Chlamydomonas_reinhardtii         None   \n",
       "29  Zea mays cantatadb                  Glycine_max         None   \n",
       "30  Zea mays cantatadb               Musa_acuminata         None   \n",
       "31  Zea mays cantatadb               Prunus_persica         None   \n",
       "32  Zea mays cantatadb  Oryza_sativa_Japonica_Group         None   \n",
       "33  Zea mays cantatadb              Sorghum_bicolor         None   \n",
       "34  Zea mays cantatadb   Selaginella_moellendorffii         None   \n",
       "35  Zea mays cantatadb        Galdieria_sulphuraria         None   \n",
       "36  Zea mays cantatadb        Physcomitrella_patens         None   \n",
       "37  Zea mays cantatadb             Chondrus_crispus         None   \n",
       "\n",
       "    conteo_lncRNA conteo_lincRNA conteo_pct  \n",
       "0           17526           None       None  \n",
       "1           12010           None       None  \n",
       "2           10924           None       None  \n",
       "3           10404           None       None  \n",
       "4           10261           None       None  \n",
       "5           10179           None       None  \n",
       "6            9504           None       None  \n",
       "7            8955           None       None  \n",
       "8            8501           None       None  \n",
       "9            7970           None       None  \n",
       "10           7593           None       None  \n",
       "11           7348           None       None  \n",
       "12           7338           None       None  \n",
       "13           7062           None       None  \n",
       "14           6459           None       None  \n",
       "15           6402           None       None  \n",
       "16           6004           None       None  \n",
       "17           5790           None       None  \n",
       "18           5511           None       None  \n",
       "19           5256           None       None  \n",
       "20           4945           None       None  \n",
       "21           4716           None       None  \n",
       "22           4542           None       None  \n",
       "23           4373           None       None  \n",
       "24           4322           None       None  \n",
       "25           4208           None       None  \n",
       "26           3590           None       None  \n",
       "27           3459           None       None  \n",
       "28           3425           None       None  \n",
       "29           3096           None       None  \n",
       "30           3001           None       None  \n",
       "31           2902           None       None  \n",
       "32           2788           None       None  \n",
       "33           2600           None       None  \n",
       "34           2267           None       None  \n",
       "35           1917           None       None  \n",
       "36           1498           None       None  \n",
       "37            224           None       None  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import os\n",
    "import mysql.connector\n",
    "import pandas as pd\n",
    "\n",
    "#método para leer los primeros num transcriptomas de un archivo fasta en cantatadb\n",
    "def num_transcriptomas_cantatadb(archivo):\n",
    "    num = 0\n",
    "    f = open(archivo, \"r\")\n",
    "    for linea in f:\n",
    "        if linea.startswith(\">\"):\n",
    "            num += 1\n",
    "    return num\n",
    "\n",
    "conn = mysql.connector.connect(\n",
    "   host=\"localhost\",\n",
    "   database=\"tesis2\",\n",
    "   user=\"root\",\n",
    "   passwd=\"\"\n",
    ")\n",
    "\n",
    "cur = conn.cursor()\n",
    "cur.execute(\"DELETE FROM conteo_lncrna_especie WHERE para_especie = 'Zea mays cantatadb'\")\n",
    "conn.commit()\n",
    "\n",
    "directory = '.\\\\httpcantata.amu.edu.pldownload.php\\\\Otros'\n",
    "\n",
    "for filename in os.listdir(directory):\n",
    "    if filename.endswith(\".fasta\"):\n",
    "        especie = os.path.splitext(filename)[0].split(\"_lncrnas\")[:-1][0]\n",
    "        query = \"INSERT INTO conteo_lncrna_especie (para_especie, especie, conteo_lncrna) VALUES (%s, %s, %s)\"\n",
    "        conteo = num_transcriptomas_cantatadb(directory + \"\\\\\" + filename)\n",
    "        cur.execute(query, ('Zea mays cantatadb', especie, conteo))\n",
    "conn.commit()\n",
    "\n",
    "df = pd.read_sql_query(\"select * from conteo_lncrna_especie WHERE para_especie = 'Zea mays cantatadb' order by 4 desc\", conn)\n",
    "display(df)\n",
    "\n",
    "conn.close()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
